{
 "cells": [
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2025-11-06T20:16:28.353424Z",
     "start_time": "2025-11-06T20:16:24.623474Z"
    }
   },
   "source": [
    "from transformers import AutoProcessor, AutoModelForCausalLM, AutoTokenizer\n",
    "\n",
    "model_name = \"internlm/Intern-S1-FP8\"\n",
    "tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)\n",
    "tokenizer.tokenize(':(A)')"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[':(', 'A', ')']"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 1
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-28T04:43:39.305502Z",
     "start_time": "2025-10-28T04:43:39.300443Z"
    }
   },
   "cell_type": "code",
   "source": "tokenizer.tokenize('\\n(A)')",
   "id": "f56deebee4bcbdda",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Ċ', '(A', ')']"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 2
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-04T02:43:41.377964Z",
     "start_time": "2025-11-04T02:43:41.371513Z"
    }
   },
   "cell_type": "code",
   "source": [
    "record = {\n",
    "    \"instruction\": \"计算这些物品的总费用。\",\n",
    "    \"input\": \"汽车-3000，衣服-100，书-20\",\n",
    "    \"output\": \"总费用为 3120。\"\n",
    "}\n",
    "user_text = (record[\"instruction\"] + (\"\\n\" + record[\"input\"] if record.get(\"input\") else \"\")).strip()\n",
    "messages = [\n",
    "    {\"role\": \"system\", \"content\": record.get(\"system\", \"\")},   # 可无\n",
    "    {\"role\": \"user\",   \"content\": user_text},\n",
    "    {\"role\": \"assistant\", \"content\": record[\"output\"]}         # 训练时 labels 只覆盖回答部分\n",
    "]\n",
    "rendered = tokenizer.apply_chat_template(\n",
    "    messages,\n",
    "    tokenize=False,\n",
    "    enable_thinking=False,\n",
    "    add_generation_prompt=True   # 训练样本里通常不加“等待助手回复”的尾部\n",
    ")\n",
    "print(\"===Rendered text===\")\n",
    "print(rendered)\n",
    "print(\"===Rendered text===\")"
   ],
   "id": "972a4af9b45fc1d5",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===Rendered text===\n",
      "<|im_start|>user\n",
      "计算这些物品的总费用。\n",
      "汽车-3000，衣服-100，书-20<|im_end|>\n",
      "<|im_start|>assistant\n",
      "<think>\n",
      "\n",
      "</think>\n",
      "总费用为 3120。<|im_end|>\n",
      "<|im_start|>assistant\n",
      "\n",
      "===Rendered text===\n"
     ]
    }
   ],
   "execution_count": 4
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-23T03:11:06.498693Z",
     "start_time": "2025-10-23T03:11:06.491958Z"
    }
   },
   "cell_type": "code",
   "source": [
    "record = {\n",
    "    \"instruction\": \"计算这些物品的总费用。\",\n",
    "    \"input\": \"汽车-3000，衣服-100，书-20\",\n",
    "    \"output\": \"总费用为 3120。\"\n",
    "}\n",
    "# conv = [{\"role\": \"user\", \"content\": [{\"type\": \"text\", \"text\": text}]}]\n",
    "user_text = (record[\"instruction\"] + (\"\\n\" + record[\"input\"] if record.get(\"input\") else \"\")).strip()\n",
    "messages = [\n",
    "    {\"role\": \"system\", \"content\": record.get(\"system\", \"\")},   # 可无\n",
    "    {\"role\": \"user\",   \"content\": [{\"type\": \"text\", \"text\": user_text}]},\n",
    "    {\"role\": \"assistant\", \"content\": [{\"type\": \"text\", \"text\": record[\"output\"]}]}         # 训练时 labels 只覆盖回答部分\n",
    "]\n",
    "rendered = tokenizer.apply_chat_template(\n",
    "    messages,\n",
    "    tokenize=False,\n",
    "    enable_thinking=False,\n",
    "    add_generation_prompt=False   # 训练样本里通常不加“等待助手回复”的尾部\n",
    ")\n",
    "print(\"===Rendered text===\")\n",
    "print(rendered)"
   ],
   "id": "c804d260c7d59442",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===Rendered text===\n",
      "<|im_start|>user\n",
      "计算这些物品的总费用。\n",
      "汽车-3000，衣服-100，书-20<|im_end|>\n",
      "<|im_start|>assistant\n",
      "<think>\n",
      "\n",
      "</think>\n",
      "总费用为 3120。<|im_end|>\n",
      "\n"
     ]
    }
   ],
   "execution_count": 3
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-17T12:59:19.977051Z",
     "start_time": "2025-10-17T12:59:19.972449Z"
    }
   },
   "cell_type": "code",
   "source": "tokenizer.additional_special_tokens",
   "id": "3c71f4d4a57efcab",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['<|im_start|>',\n",
       " '<|im_end|>',\n",
       " '<|object_ref_start|>',\n",
       " '<|object_ref_end|>',\n",
       " '<|box_start|>',\n",
       " '<|box_end|>',\n",
       " '<|quad_start|>',\n",
       " '<|quad_end|>',\n",
       " '<|vision_start|>',\n",
       " '<|vision_end|>',\n",
       " '<|vision_pad|>',\n",
       " '<|image_pad|>',\n",
       " '<|video_pad|>']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 7
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-17T12:59:43.639231Z",
     "start_time": "2025-10-17T12:59:43.634456Z"
    }
   },
   "cell_type": "code",
   "source": "tokenizer.all_special_tokens",
   "id": "28a537bad6503395",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['<|im_end|>',\n",
       " '<|endoftext|>',\n",
       " '<|im_start|>',\n",
       " '<|object_ref_start|>',\n",
       " '<|object_ref_end|>',\n",
       " '<|box_start|>',\n",
       " '<|box_end|>',\n",
       " '<|quad_start|>',\n",
       " '<|quad_end|>',\n",
       " '<|vision_start|>',\n",
       " '<|vision_end|>',\n",
       " '<|vision_pad|>',\n",
       " '<|image_pad|>',\n",
       " '<|video_pad|>',\n",
       " '<IMG_CONTEXT>',\n",
       " '</img>',\n",
       " '<img>',\n",
       " '<video>']"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 9
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-17T13:00:24.552045Z",
     "start_time": "2025-10-17T13:00:24.546789Z"
    }
   },
   "cell_type": "code",
   "source": "tokenizer.get_added_vocab()",
   "id": "9fe8f8a454d9c985",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'<|endoftext|>': 151643,\n",
       " '<|im_start|>': 151644,\n",
       " '<|im_end|>': 151645,\n",
       " '<|object_ref_start|>': 151646,\n",
       " '<|object_ref_end|>': 151647,\n",
       " '<|box_start|>': 151648,\n",
       " '<|box_end|>': 151649,\n",
       " '<|quad_start|>': 151650,\n",
       " '<|quad_end|>': 151651,\n",
       " '<|vision_start|>': 151652,\n",
       " '<|vision_end|>': 151653,\n",
       " '<|vision_pad|>': 151654,\n",
       " '<|image_pad|>': 151655,\n",
       " '<|video_pad|>': 151656,\n",
       " '<tool_call>': 151657,\n",
       " '</tool_call>': 151658,\n",
       " '<|fim_prefix|>': 151659,\n",
       " '<|fim_middle|>': 151660,\n",
       " '<|fim_suffix|>': 151661,\n",
       " '<|fim_pad|>': 151662,\n",
       " '<|repo_name|>': 151663,\n",
       " '<|file_sep|>': 151664,\n",
       " '<tool_response>': 151665,\n",
       " '</tool_response>': 151666,\n",
       " '<think>': 151667,\n",
       " '</think>': 151668,\n",
       " '<SMILES>': 151925,\n",
       " '</SMILES>': 151926,\n",
       " '<SELFIES>': 151927,\n",
       " '</SELFIES>': 151928,\n",
       " '<IUPAC>': 152441,\n",
       " '</IUPAC>': 152442,\n",
       " '<FASTA>': 152955,\n",
       " '</FASTA>': 152956,\n",
       " '<IMG_CONTEXT>': 152957,\n",
       " '<img>': 152958,\n",
       " '</img>': 152959,\n",
       " '<quad>': 152960,\n",
       " '</quad>': 152961,\n",
       " '<ref>': 152962,\n",
       " '</ref>': 152963,\n",
       " '<box>': 152964,\n",
       " '</box>': 152965,\n",
       " '<|action_start|>': 152966,\n",
       " '<|action_end|>': 152967,\n",
       " '<|interpreter|>': 152968,\n",
       " '<|plugin|>': 152969,\n",
       " '<video>': 152970}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 11
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-08T03:53:47.509581Z",
     "start_time": "2025-10-08T03:53:47.505181Z"
    }
   },
   "cell_type": "code",
   "source": "tokenizer.tokenize('A**')",
   "id": "dfafb216ea0e27f",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['A', '**']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 11
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-07T21:03:12.433752Z",
     "start_time": "2025-10-07T21:03:12.427017Z"
    }
   },
   "cell_type": "code",
   "source": "tokenizer.tokenize('<|im_start|>user\\nInstructions: Answer the following question about drug properties.\\nContext: As a membrane separating circulating blood and brain extracellular fluid, the blood-brain barrier (BBB) is the protection layer that blocks most foreign drugs. Thus the ability of a drug to penetrate the barrier to deliver to the site of action forms a crucial challenge in development of drugs for central nervous system.\\nQuestion: Given a drug SMILES string, predict whether it\\n(A) does not cross the BBB (B) crosses the BBB\\nDrug SMILES: <SMILES>CC[C@H](C)C(=O)O[C@H]1CCC=C2C=C[C@H](C)[C@H](CC[C@@H]3C[C@@H](O)CC(=O)O3)[C@H]21</SMILES>\\nAnswer:<|im_end|>\\n<|im_start|>assistant\\n(A')[142]",
   "id": "b34e79f0bba97d64",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'(A'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 12
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-20T21:28:45.616354Z",
     "start_time": "2025-10-20T21:28:45.454546Z"
    }
   },
   "cell_type": "code",
   "source": [
    "from tdc.single_pred import ADME\n",
    "from tdc.multi_pred import PPI"
   ],
   "id": "c6fb17a56a074637",
   "outputs": [
    {
     "ename": "ImportError",
     "evalue": "cannot import name 'AffineTransform' from 'somacore' (/home/tianang/anaconda3/envs/reasonv/lib/python3.11/site-packages/somacore/__init__.py)",
     "output_type": "error",
     "traceback": [
      "\u001B[31m---------------------------------------------------------------------------\u001B[39m",
      "\u001B[31mImportError\u001B[39m                               Traceback (most recent call last)",
      "\u001B[36mCell\u001B[39m\u001B[36m \u001B[39m\u001B[32mIn[14]\u001B[39m\u001B[32m, line 2\u001B[39m\n\u001B[32m      1\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01mtdc\u001B[39;00m\u001B[34;01m.\u001B[39;00m\u001B[34;01msingle_pred\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m ADME\n\u001B[32m----> \u001B[39m\u001B[32m2\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01mtdc\u001B[39;00m\u001B[34;01m.\u001B[39;00m\u001B[34;01mmulti_pred\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m PPI\n",
      "\u001B[36mFile \u001B[39m\u001B[32m~/anaconda3/envs/reasonv/lib/python3.11/site-packages/tdc/multi_pred/__init__.py:15\u001B[39m\n\u001B[32m     13\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01m.\u001B[39;00m\u001B[34;01mtcr_epi\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m TCREpitopeBinding\n\u001B[32m     14\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01m.\u001B[39;00m\u001B[34;01mtrialoutcome\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m TrialOutcome\n\u001B[32m---> \u001B[39m\u001B[32m15\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01m.\u001B[39;00m\u001B[34;01mperturboutcome\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m PerturbOutcome\n",
      "\u001B[36mFile \u001B[39m\u001B[32m~/anaconda3/envs/reasonv/lib/python3.11/site-packages/tdc/multi_pred/perturboutcome.py:13\u001B[39m\n\u001B[32m     10\u001B[39m \u001B[38;5;28;01mimport\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01msys\u001B[39;00m\n\u001B[32m     12\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01m.\u001B[39;00m\u001B[34;01m.\u001B[39;00m\u001B[34;01mutils\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m print_sys\n\u001B[32m---> \u001B[39m\u001B[32m13\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01m.\u001B[39;00m\u001B[34;01msingle_cell\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m CellXGeneTemplate\n\u001B[32m     14\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01m.\u001B[39;00m\u001B[34;01m.\u001B[39;00m\u001B[34;01mdataset_configs\u001B[39;00m\u001B[34;01m.\u001B[39;00m\u001B[34;01mconfig_map\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m scperturb_gene_datasets\n\u001B[32m     17\u001B[39m \u001B[38;5;28;01mdef\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34mparse_single_pert\u001B[39m(i):\n",
      "\u001B[36mFile \u001B[39m\u001B[32m~/anaconda3/envs/reasonv/lib/python3.11/site-packages/tdc/multi_pred/single_cell.py:3\u001B[39m\n\u001B[32m      1\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01mcollections\u001B[39;00m\u001B[34;01m.\u001B[39;00m\u001B[34;01mabc\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m Iterable\n\u001B[32m      2\u001B[39m \u001B[38;5;28;01mimport\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01mpandas\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mas\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01mpd\u001B[39;00m\n\u001B[32m----> \u001B[39m\u001B[32m3\u001B[39m \u001B[38;5;28;01mimport\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01mtiledbsoma\u001B[39;00m\n\u001B[32m      5\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01m.\u001B[39;00m\u001B[34;01manndata_dataset\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m DataLoader\n\u001B[32m      6\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01m.\u001B[39;00m\u001B[34;01m.\u001B[39;00m\u001B[34;01mdataset_configs\u001B[39;00m\u001B[34;01m.\u001B[39;00m\u001B[34;01mcellxgene_config\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m CellXGeneConfig\n",
      "\u001B[36mFile \u001B[39m\u001B[32m~/anaconda3/envs/reasonv/lib/python3.11/site-packages/tiledbsoma/__init__.py:142\u001B[39m\n\u001B[32m    137\u001B[39m     \u001B[38;5;28;01mexcept\u001B[39;00m \u001B[38;5;167;01mOSError\u001B[39;00m:\n\u001B[32m    138\u001B[39m         \u001B[38;5;66;03m# Otherwise try loading by name only.\u001B[39;00m\n\u001B[32m    139\u001B[39m         ctypes.CDLL(libtiledbsoma_name)\n\u001B[32m--> \u001B[39m\u001B[32m142\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01msomacore\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m (\n\u001B[32m    143\u001B[39m     AffineTransform,\n\u001B[32m    144\u001B[39m     Axis,\n\u001B[32m    145\u001B[39m     AxisColumnNames,\n\u001B[32m    146\u001B[39m     AxisQuery,\n\u001B[32m    147\u001B[39m     CoordinateSpace,\n\u001B[32m    148\u001B[39m     IdentityTransform,\n\u001B[32m    149\u001B[39m     ScaleTransform,\n\u001B[32m    150\u001B[39m     UniformScaleTransform,\n\u001B[32m    151\u001B[39m )\n\u001B[32m    152\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01msomacore\u001B[39;00m\u001B[34;01m.\u001B[39;00m\u001B[34;01moptions\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m ResultOrder\n\u001B[32m    154\u001B[39m \u001B[38;5;66;03m# This is important since we need to do the above dll/dylib/so business\u001B[39;00m\n\u001B[32m    155\u001B[39m \u001B[38;5;66;03m# _before_ imports, but, ruff will tell us that imports need to be\u001B[39;00m\n\u001B[32m    156\u001B[39m \u001B[38;5;66;03m# at the top of the file:\u001B[39;00m\n\u001B[32m    157\u001B[39m \u001B[38;5;66;03m#\u001B[39;00m\n",
      "\u001B[31mImportError\u001B[39m: cannot import name 'AffineTransform' from 'somacore' (/home/tianang/anaconda3/envs/reasonv/lib/python3.11/site-packages/somacore/__init__.py)"
     ]
    }
   ],
   "execution_count": 14
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-30T05:38:50.136543Z",
     "start_time": "2025-10-30T05:38:42.953603Z"
    }
   },
   "cell_type": "code",
   "source": [
    "from tdc.single_pred import ADME, Tox, QM, Epitope, Develop, CRISPROutcome, Yields\n",
    "from tdc.multi_pred.ppi import PPI#, TCREpitopeBinding, TrialOutcome, PeptideMHC\n",
    "from tdc.multi_pred.tcr_epi import TCREpitopeBinding\n",
    "from tdc.multi_pred.trialoutcome import TrialOutcome\n",
    "from tdc.multi_pred.peptidemhc import PeptideMHC\n",
    "from tdc.multi_pred.dti import DTI\n",
    "from tdc.multi_pred.drugsyn import DrugSyn\n",
    "from tdc.multi_pred.drugres import DrugRes\n",
    "from tdc.multi_pred.antibodyaff import AntibodyAff\n",
    "from huggingface_hub import hf_hub_download\n",
    "import json\n",
    "import numpy as np\n",
    "from tdc.utils import retrieve_label_name_list\n",
    "print(1)\n",
    "\n",
    "tdc_prompts_filepath = hf_hub_download(\n",
    "    repo_id=\"google/txgemma-9b-predict\",\n",
    "    filename=\"tdc_prompts.json\",\n",
    ")\n",
    "tdc_prompts_json = json.load(open(tdc_prompts_filepath))\n",
    "\n",
    "\n",
    "# 选择一个任务和输入\n",
    "task_name = \"CYP1A2_Veith\"  # BBB_Martins\n",
    "# label_list = retrieve_label_name_list(task_name)\n",
    "# print(label_list)\n",
    "input_type = \"{Drug SMILES}\"\n",
    "\n",
    "# SFT_data = Tox(name=task_name, label_name = label_list[2])\n",
    "data = ADME(name=task_name)\n",
    "# SFT_data = SFT_data.neg_sample(frac = 1)  # only for PPI dataset\n",
    "split = data.get_split()\n",
    "print(split['test'])\n",
    "# print((split['test']['Y'].to_numpy()).sum())#[:5]\n",
    "# print(split['test']['Antibody'].values)\n",
    "test_drugs = split['test']['Drug']#[:5]\n",
    "test_labels = split['test']['Y']#[:5]\n",
    "\n",
    "drug_smiles = test_drugs[0]\n",
    "\n",
    "prompt_text = tdc_prompts_json[task_name].replace(input_type, f'<SMILES>{drug_smiles}</SMILES>')\n",
    "print(prompt_text)\n",
    "conv = [{\"role\": \"user\", \"content\": [{\"type\": \"text\", \"text\": prompt_text}]}]\n",
    "tokenizer.apply_chat_template(\n",
    "            conv,\n",
    "            tokenize=False,  # 让 vLLM 自己分词\n",
    "            add_generation_prompt=True,  # 补上assistant起始标记\n",
    "            enable_thinking=False\n",
    "        )"
   ],
   "id": "f3d19d62f9094d2",
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Downloading...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 760k/760k [00:00<00:00, 2.45MiB/s]\n",
      "Loading...\n",
      "Done!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Drug_ID                                               Drug  Y\n",
      "0       700531.0        O=C(O)[C@H](Cc1cnc[nH]1)N1C(=O)c2ccccc2C1=O  0\n",
      "1      6270538.0            COc1ccc(CNC(=O)/C=C/c2ccc(OCCCF)cc2)cc1  0\n",
      "2      3239491.0                            CC(=O)NC(O)C(=O)c1ccco1  0\n",
      "3      5771786.0               CCCC/C=C/C(NC(=O)CCc1ccccc1)c1ccccc1  1\n",
      "4       738566.0                      COc1cccc(CNCc2ccc(SC)cc2)c1OC  1\n",
      "...          ...                                                ... ..\n",
      "2511   3435952.0             O=C(Nc1nnc(SCc2c(Cl)cccc2Cl)s1)C1CCCO1  1\n",
      "2512   3246557.0                    O=C(O)[C@@H](Br)[C@H](Br)C(=O)O  0\n",
      "2513  16758750.0  COc1ccc(COC(=O)N/N=C2/C[C@@H](O)[C@@H](O)[C@@H...  0\n",
      "2514    135349.0  CN(C(=O)Cc1ccc(Cl)c(Cl)c1)[C@@H]1CCCC[C@H]1N1C...  0\n",
      "2515   2295127.0  Cc1ccc(S(=O)(=O)Oc2ccc(C3C4=C(CCCC4=O)OC4=C3C(...  0\n",
      "\n",
      "[2516 rows x 3 columns]\n",
      "Instructions: Answer the following question about drug properties.\n",
      "Context: The CYP P450 genes are involved in the formation and breakdown (metabolism) of various molecules and chemicals within cells. Specifically, CYP1A2 localizes to the endoplasmic reticulum and its expression is induced by some polycyclic aromatic hydrocarbons (PAHs), some of which are found in cigarette smoke. It is able to metabolize some PAHs to carcinogenic intermediates. Other xenobiotic substrates for this enzyme include caffeine, aflatoxin B1, and acetaminophen.\n",
      "Question: Given a drug SMILES string, predict whether it\n",
      "(A) does not inhibit CYP1A2 (B) inhibits CYP1A2\n",
      "Drug SMILES: <SMILES>O=C(O)[C@H](Cc1cnc[nH]1)N1C(=O)c2ccccc2C1=O</SMILES>\n",
      "Answer:\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'tokenizer' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001B[31m---------------------------------------------------------------------------\u001B[39m",
      "\u001B[31mNameError\u001B[39m                                 Traceback (most recent call last)",
      "\u001B[36mCell\u001B[39m\u001B[36m \u001B[39m\u001B[32mIn[1]\u001B[39m\u001B[32m, line 44\u001B[39m\n\u001B[32m     42\u001B[39m \u001B[38;5;28mprint\u001B[39m(prompt_text)\n\u001B[32m     43\u001B[39m conv = [{\u001B[33m\"\u001B[39m\u001B[33mrole\u001B[39m\u001B[33m\"\u001B[39m: \u001B[33m\"\u001B[39m\u001B[33muser\u001B[39m\u001B[33m\"\u001B[39m, \u001B[33m\"\u001B[39m\u001B[33mcontent\u001B[39m\u001B[33m\"\u001B[39m: [{\u001B[33m\"\u001B[39m\u001B[33mtype\u001B[39m\u001B[33m\"\u001B[39m: \u001B[33m\"\u001B[39m\u001B[33mtext\u001B[39m\u001B[33m\"\u001B[39m, \u001B[33m\"\u001B[39m\u001B[33mtext\u001B[39m\u001B[33m\"\u001B[39m: prompt_text}]}]\n\u001B[32m---> \u001B[39m\u001B[32m44\u001B[39m \u001B[43mtokenizer\u001B[49m.apply_chat_template(\n\u001B[32m     45\u001B[39m             conv,\n\u001B[32m     46\u001B[39m             tokenize=\u001B[38;5;28;01mFalse\u001B[39;00m,  \u001B[38;5;66;03m# 让 vLLM 自己分词\u001B[39;00m\n\u001B[32m     47\u001B[39m             add_generation_prompt=\u001B[38;5;28;01mTrue\u001B[39;00m,  \u001B[38;5;66;03m# 补上assistant起始标记\u001B[39;00m\n\u001B[32m     48\u001B[39m             enable_thinking=\u001B[38;5;28;01mFalse\u001B[39;00m\n\u001B[32m     49\u001B[39m         )\n",
      "\u001B[31mNameError\u001B[39m: name 'tokenizer' is not defined"
     ]
    }
   ],
   "execution_count": 1
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-23T05:01:51.575202Z",
     "start_time": "2025-10-23T05:01:51.568566Z"
    }
   },
   "cell_type": "code",
   "source": "tdc_prompts_json['MHC1_IEDB_IMGT_Nielsen']",
   "id": "9d8695fc112683d2",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Instructions: Answer the following question about peptide-MHC binding.\\nContext: In the human body, T cells monitor the existing peptides and trigger an immune response if the peptide is foreign. To decide whether or not if the peptide is not foreign, the peptide must bind to a major histocompatibility complex (MHC) molecule. Therefore, predicting peptide-MHC binding affinity is pivotal for determining immunogenicity. In some experiments, the peptide binding is measured against cells that express multiple MHCs, so the peptide could be binding any one of the possible MHCs. Class 1 MHC molecules bind to peptides that are usually 8-14 amino acids long and activate CD8 T cells.\\nQuestion: Given the amino acid sequence of the peptide and possible pseudo amino acid sequences of MHC 1, predict whether the peptide\\n(A) does not bind to any of the MHCs (B) binds to any of the MHCs\\nPeptide amino acid sequence: {Peptide amino acid sequence}\\nPossible MHC pseudosequences: {Possible MHC pseudosequences}\\nAnswer:'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 3
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### 回归任务的模版",
   "id": "90bae7d3764f40f9"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-08T03:54:49.183198Z",
     "start_time": "2025-10-08T03:54:49.177799Z"
    }
   },
   "cell_type": "code",
   "source": [
    "for i, template in tdc_prompts_json.items():\n",
    "    if '(A)' not in template:\n",
    "        print(f'task {i}:\\n'+template+'\\n')"
   ],
   "id": "2b6f05bc3445d2e9",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "task Caco2_Wang:\n",
      "Instructions: Answer the following question about drug properties.\n",
      "Context: The human colon epithelial cancer cell line, Caco-2, is used as an in vitro model to simulate the human intestinal tissue. The experimental result on the rate of drug passing through the Caco-2 cells can approximate the rate at which the drug permeates through the human intestinal tissue.\n",
      "Question: Given a drug SMILES string, predict its normalized Caco-2 cell effective permeability from 000 to 1000, where 000 is minimum permeability and 1000 is maximum permeability.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Answer:\n",
      "\n",
      "task VDss_Lombardo:\n",
      "Instructions: Answer the following question about drug properties.\n",
      "Context: The volume of distribution at steady state (VDss) measures the degree of a drug's concentration in body tissue compared to concentration in blood. Higher VD indicates a higher distribution in the tissue and usually indicates the drug with high lipid solubility, low plasma protein binding rate.\n",
      "Question: Given a drug SMILES string, predict its normalized volume of distribution from 000 to 1000, where 000 is minimum volume of distribution and 1000 is maximum volume of distribution.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Answer:\n",
      "\n",
      "task Half_Life_Obach:\n",
      "Instructions: Answer the following question about drug properties.\n",
      "Context: Half life of a drug is the duration for the concentration of the drug in the body to be reduced by half. It measures the duration of actions of a drug. \n",
      "Question: Given a drug SMILES string, predict its normalized half life from 000 to 1000, where 000 is minimum half life and 1000 is maximum half life.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Answer:\n",
      "\n",
      "task Clearance_Hepatocyte_AZ:\n",
      "Instructions: Answer the following question about drug properties.\n",
      "Context: Drug clearance is defined as the volume of plasma cleared of a drug over a specified time period and it measures the rate at which the active drug is removed from the body.\n",
      "Question: Given a drug SMILES string, predict its normalized hepatocyte clearance from 000 to 1000, where 000 is minimum hepatocyte clearance and 1000 is maximum hepatocyte clearance.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Answer:\n",
      "\n",
      "task Clearance_Microsome_AZ:\n",
      "Instructions: Answer the following question about drug properties.\n",
      "Context: Drug clearance is defined as the volume of plasma cleared of a drug over a specified time period and it measures the rate at which the active drug is removed from the body.\n",
      "Question: Given a drug SMILES string, predict its normalized microsome clearance activity from 000 to 1000, where 000 is minimum microsome clearance and 1000 is maximum microsome clearance.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Answer:\n",
      "\n",
      "task LD50_Zhu:\n",
      "Instructions: Answer the following question about drug properties.\n",
      "Context: Acute toxicity LD50 measures the most conservative dose that can lead to lethal adverse effects. The lower the dose, the more lethal of a drug. \n",
      "Question: Given a drug SMILES string, predict its normalized LD50 from 000 to 1000, where 000 is minimum LD50 and 1000 is maximum LD50.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Answer:\n",
      "\n",
      "task PPBR_AZ:\n",
      "Instructions: Answer the following question about drug properties.\n",
      "Context: The human plasma protein binding rate (PPBR) is expressed as the percentage of a drug bound to plasma proteins in the blood. This rate strongly affect a drug's efficiency of delivery. The less bound a drug is, the more efficiently it can traverse and diffuse to the site of actions.\n",
      "Question: Given a drug SMILES string, predict its normalized rate of PPBR from 000 to 1000, where 000 is minimum PPBR rate and 1000 is maximum PPBR rate.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Answer:\n",
      "\n",
      "task BindingDB_kd:\n",
      "Instructions: Answer the following question about drug target interactions.\n",
      "Context: Drug-target binding is the physical interaction between a drug and a specific biological molecule, such as a protein or enzyme. This interaction is essential for the drug to exert its pharmacological effect. The strength of the drug-target binding is determined by the binding affinity, which is a measure of how tightly the drug binds to the target. Kd is the dissociation constant of a drug-target complex. It is the concentration of drug at which half of the drug-target complexes have dissociated. A lower Kd value indicates a stronger binding affinity.\n",
      "Question: Given the target amino acid sequence and compound SMILES string, predict their normalized binding affinity Kd from 000 to 1000, where 000 is minimum Kd and 1000 is maximum Kd.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Target amino acid sequence: {Target amino acid sequence}\n",
      "Answer:\n",
      "\n",
      "task BindingDB_ic50:\n",
      "Instructions: Answer the following question about drug target interactions.\n",
      "Context: Drug-target binding is the physical interaction between a drug and a specific biological molecule, such as a protein or enzyme. This interaction is essential for the drug to exert its pharmacological effect. The strength of the drug-target binding is determined by the binding affinity, which is a measure of how tightly the drug binds to the target. IC50 is the concentration of a drug that inhibits a biological activity by 50%. It is a measure of the drug's potency, but it is not a direct measure of binding affinity. \n",
      "Question: Given the target amino acid sequence and compound SMILES string, predict their normalized binding affinity Kd from 000 to 1000, where 000 is minimum IC50 and 1000 is maximum IC50.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Target amino acid sequence: {Target amino acid sequence}\n",
      "Answer:\n",
      "\n",
      "task BindingDB_ki:\n",
      "Instructions: Answer the following question about drug target interactions.\n",
      "Context: Drug-target binding is the physical interaction between a drug and a specific biological molecule, such as a protein or enzyme. This interaction is essential for the drug to exert its pharmacological effect. The strength of the drug-target binding is determined by the binding affinity, which is a measure of how tightly the drug binds to the target. Ki is the equilibrium dissociation constant of an inhibitor. It is the concentration of inhibitor at which half of the target binding sites are occupied. A lower Ki value indicates a stronger binding affinity.\n",
      "Question: Given the target amino acid sequence and compound SMILES string, predict their normalized binding affinity Kd from 000 to 1000, where 000 is minimum Ki and 1000 is maximum Ki.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Target amino acid sequence: {Target amino acid sequence}\n",
      "Answer:\n",
      "\n",
      "task DAVIS:\n",
      "Instructions: Answer the following question about drug target interactions.\n",
      "Context: Studying the interaction between kinase inhibitors and kinases is crucial for understanding complex cellular processes, advancing drug development, enabling targeted cancer therapies, and addressing drug resistance. This research also aids in biomarker identification for personalized medicine, offers potential treatments for neurological and other diseases, and contributes significantly to the broader understanding of kinase biology and enzymology. The strength of the drug-target binding is determined by the binding affinity, which is a measure of how tightly the drug binds to the target. Kd is the dissociation constant of a drug-target complex. It is the concentration of drug at which half of the drug-target complexes have dissociated. A lower Kd value indicates a stronger binding affinity. \n",
      "Question: Given the target kinase amino acid sequence and kinase inhibitor SMILES string, predict their normalized binding affinity Kd from 000 to 1000, where 000 is minimum Kd and 1000 is maximum Kd.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Target amino acid sequence: {Target amino acid sequence}\n",
      "Answer:\n",
      "\n",
      "task USPTO_Yields:\n",
      "Instructions: Answer the following question about reaction yields.\n",
      "Context: Vast majority of small-molecule drugs are synthesized through chemical reactions. Many factors during reactions could lead to suboptimal reactants-products conversion rate, i.e. yields. Formally, it is defined as the percentage of the reactants successfully converted to the target product. Yields were measured for various patented reactions.\n",
      "Question: Given a catalyst SMILES string, reactant SMILES string, and product SMILES string, predict the normalized yield from 000 to 1000, where 000 is minimum yield and 1000 is maximum yield.\n",
      "Catalyst SMILES: {Catalyst SMILES}\n",
      "Reactant SMILES: {Reactant SMILES}\n",
      "Product SMILES: {Product SMILES}\n",
      "Answer:\n",
      "\n",
      "task Buchwald_Hartwig:\n",
      "Instructions: Answer the following question about reaction yields.\n",
      "Context: Vast majority of small-molecule drugs are synthesized through chemical reactions. Many factors during reactions could lead to suboptimal reactants-products conversion rate, i.e. yields. Formally, it is defined as the percentage of the reactants successfully converted to the target product. Yields were measured for various Pd-catalysed Buchwald–Hartwig C-N cross coupling reactions.\n",
      "Question: Given a product SMILES string, catalyst SMILES string, and reactant SMILES string, predict the normalized yield from 000 to 1000, where 000 is minimum yield and 1000 is maximum yield.\n",
      "Product SMILES: {Product SMILES}\n",
      "Catalyst SMILES: {Catalyst SMILES}\n",
      "Reactant SMILES: {Reactant SMILES}\n",
      "Answer:\n",
      "\n",
      "task OncoPolyPharmacology:\n",
      "Instructions: Answer the following question about drug synergy.\n",
      "Context: Drug combination therapy offers enormous potential for expanding the use of existing drugs and in improving their efficacy. Synergy is a dimensionless measure of deviation of an observed drug combination response from the expected effect of non-interaction. Drug combinations were tested against cancer cell lines.\n",
      "Question: Given two drug SMILES strings and a cell line description, predict the normalized drug synergy from 000 to 1000, where 000 is minimum drug synergy and 1000 is maximum drug synergy.\n",
      "Cell line description: {Cell line description}\n",
      "Drug1 SMILES: {Drug1 SMILES}\n",
      "Drug2 SMILES: {Drug2 SMILES}\n",
      "Answer:\n",
      "\n",
      "task DrugComb_CSS:\n",
      "Instructions: Answer the following question about drug synergy.\n",
      "Context: Drug combination therapy offers enormous potential for expanding the use of existing drugs and in improving their efficacy. Synergy is a dimensionless measure of deviation of an observed drug combination response from the expected effect of non-interaction. Drug combinations were tested against cancer cell lines. CSS measures the drug combination sensitivity and is derived using relative IC50 values of compounds and the area under their dose-response curves. \n",
      "Question: Given two drug SMILES strings and a cell line description, predict the normalized drug synergy from 000 to 1000, where 000 is minimum drug synergy and 1000 is maximum drug synergy.\n",
      "Drug1 SMILES: {Drug1 SMILES}\n",
      "Drug2 SMILES: {Drug2 SMILES}\n",
      "Cell line description: {Cell line description}\n",
      "Answer:\n",
      "\n",
      "task DrugComb_HSA:\n",
      "Instructions: Answer the following question about drug synergy.\n",
      "Context: Drug combination therapy offers enormous potential for expanding the use of existing drugs and in improving their efficacy. Synergy is a dimensionless measure of deviation of an observed drug combination response from the expected effect of non-interaction. Drug combinations were tested against cancer cell lines. Synergy is calculated using Highest Single Agent (HSA).\n",
      "Question: Given two drug SMILES strings and a cell line description, predict the normalized drug synergy from 000 to 1000, where 000 is minimum drug synergy and 1000 is maximum drug synergy.\n",
      "Drug1 SMILES: {Drug1 SMILES}\n",
      "Drug2 SMILES: {Drug2 SMILES}\n",
      "Cell line description: {Cell line description}\n",
      "Answer:\n",
      "\n",
      "task DrugComb_Loewe:\n",
      "Instructions: Answer the following question about drug synergy.\n",
      "Context: Drug combination therapy offers enormous potential for expanding the use of existing drugs and in improving their efficacy. Synergy is a dimensionless measure of deviation of an observed drug combination response from the expected effect of non-interaction. Drug combinations were tested against cancer cell lines. Synergy is calculated using the Loewe additivity model.\n",
      "Question: Given two drug SMILES strings and a cell line description, predict the normalized drug synergy from 000 to 1000, where 000 is minimum drug synergy and 1000 is maximum drug synergy.\n",
      "Drug1 SMILES: {Drug1 SMILES}\n",
      "Drug2 SMILES: {Drug2 SMILES}\n",
      "Cell line description: {Cell line description}\n",
      "Answer:\n",
      "\n",
      "task DrugComb_Bliss:\n",
      "Instructions: Answer the following question about drug synergy.\n",
      "Context: Drug combination therapy offers enormous potential for expanding the use of existing drugs and in improving their efficacy. Synergy is a dimensionless measure of deviation of an observed drug combination response from the expected effect of non-interaction. Drug combinations were tested against cancer cell lines. Synergy is calculated using  the Bliss model.\n",
      "Question: Given two drug SMILES strings and a cell line description, predict the normalized drug synergy from 000 to 1000, where 000 is minimum drug synergy and 1000 is maximum drug synergy.\n",
      "Drug1 SMILES: {Drug1 SMILES}\n",
      "Drug2 SMILES: {Drug2 SMILES}\n",
      "Cell line description: {Cell line description}\n",
      "Answer:\n",
      "\n",
      "task DrugComb_ZIP:\n",
      "Instructions: Answer the following question about drug synergy.\n",
      "Context: Drug combination therapy offers enormous potential for expanding the use of existing drugs and in improving their efficacy. Synergy is a dimensionless measure of deviation of an observed drug combination response from the expected effect of non-interaction. Drug combinations were tested against cancer cell lines. Synergy is calculated using Zero Interaction Potency (ZIP).\n",
      "Question: Given two drug SMILES strings and a cell line description, predict the normalized drug synergy from 000 to 1000, where 000 is minimum drug synergy and 1000 is maximum drug synergy.\n",
      "Drug1 SMILES: {Drug1 SMILES}\n",
      "Drug2 SMILES: {Drug2 SMILES}\n",
      "Cell line description: {Cell line description}\n",
      "Answer:\n",
      "\n",
      "task Protein_SAbDab:\n",
      "Instructions: Answer the following question about antibody antigen interactions.\n",
      "Context: Antibodies recognize pathogen antigens and destroy them. The activity is measured by their binding affinities.\n",
      "Question: Given the amino acid sequence of the antibody and antigen, predict the normalized binding affinity from 000 to 1000, where 000 is minimum binding affinity and 1000 is maximum binding affinity.\n",
      "Antigen sequence: {Antigen sequence}\n",
      "Antibody heavy chain sequence: {Antibody heavy chain sequence}\n",
      "Antibody light chain sequence: {Antibody light chain sequence}\n",
      "Answer:\n",
      "\n",
      "task Lipophilicity_AstraZeneca:\n",
      "Instructions: Answer the following question about drug properties.\n",
      "Context: Lipophilicity measures the ability of a drug to dissolve in a lipid (e.g. fats, oils) environment. High lipophilicity often leads to high rate of metabolism, poor solubility, high turn-over, and low absorption.\n",
      "Question: Given a drug SMILES string, predict its normalized lipophilicity from 000 to 1000, where 000 is minimum lipophilicity and 1000 is maximum lipophilicity.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Answer:\n",
      "\n",
      "task Solubility_AqSolDB:\n",
      "Instructions: Answer the following question about drug properties.\n",
      "Context: Aqeuous solubility measures a drug's ability to dissolve in water. Poor water solubility could lead to slow drug absorptions, inadequate bioavailablity and even induce toxicity. More than 40% of new chemical entities are not soluble.\n",
      "Question: Given a drug SMILES string, predict its normalized solubility from 000 to 1000, where 000 is minimum solubility and 1000 is maximum solubility.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Answer:\n",
      "\n",
      "task TAP_CDR_Length:\n",
      "Instructions: Answer the following question about antibody developability.\n",
      "Context: Immunogenicity, instability, self-association, high viscosity, polyspecificity, or poor expression can all preclude an antibody from becoming a therapeutic. Early identification of these negative characteristics is essential. Akin to the Lipinski guidelines, which measure druglikeness in small molecules, Therapeutic Antibody Profiler (TAP) highlights antibodies that possess characteristics that are rare/unseen in clinical-stage mAb therapeutics. One key criteria is the total length of the complementarity-determining regions (CDRs). Loop length has a major impact on the nature of antigen binding. For example, if an antibody has a long CDRH3 loop, it tends to form most of the interactions with an antigen, while shorter CDRH3 loops contribute to concave binding sites where other CDRs more often assist in binding. \n",
      "Question: Given an antibody heavy chain and light chain sequence, predict its normalized CDR length from 000 to 1000, where 000 is minimum CDR length and 1000 is maximum CDR length.\n",
      "Antibody heavy chain sequence: {Antibody heavy chain sequence}\n",
      "Antibody light chain sequence: {Antibody light chain sequence}\n",
      "Answer:\n",
      "\n",
      "task TAP_PSH:\n",
      "Instructions: Answer the following question about antibody developability.\n",
      "Context: Immunogenicity, instability, self-association, high viscosity, polyspecificity, or poor expression can all preclude an antibody from becoming a therapeutic. Early identification of these negative characteristics is essential. Akin to the Lipinski guidelines, which measure druglikeness in small molecules, Therapeutic Antibody Profiler (TAP) highlights antibodies that possess characteristics that are rare/unseen in clinical-stage mAb therapeutics. One key criteria the extent and magnitude of surface hydrophobicity. Hydrophobicity in the CDR regions has been repeatedly linked to aggregation propensity in mAbs. Hydrophobicity can be measured by a PSH metric that yields higher scores if hydrophobic residues tend to neighbor one another in a region, rather than being evenly separated.\n",
      "Question: Given an antibody heavy chain and light chain sequence, predict its normalized PSH from 000 to 1000, where 000 is minimum PSH and 1000 is maximum PSH.\n",
      "Antibody heavy chain sequence: {Antibody heavy chain sequence}\n",
      "Antibody light chain sequence: {Antibody light chain sequence}\n",
      "Answer:\n",
      "\n",
      "task TAP_PPC:\n",
      "Instructions: Answer the following question about antibody developability.\n",
      "Context: Immunogenicity, instability, self-association, high viscosity, polyspecificity, or poor expression can all preclude an antibody from becoming a therapeutic. Early identification of these negative characteristics is essential. Akin to the Lipinski guidelines, which measure druglikeness in small molecules, Therapeutic Antibody Profiler (TAP) highlights antibodies that possess characteristics that are rare/unseen in clinical-stage mAb therapeutics. One key criteria is positive charge and negative charge in the CDRs. Surface patches of positive or negative charge in the CDRs are linked to high rates of clearance and poor expression levels. PPC is a score highlighting regions of dense positive charge.\n",
      "Question: Given an antibody heavy chain and light chain sequence, predict its normalized PPC score from 000 to 1000, where 000 is minimum PPC and 1000 is maximum PPC.\n",
      "Antibody heavy chain sequence: {Antibody heavy chain sequence}\n",
      "Antibody light chain sequence: {Antibody light chain sequence}\n",
      "Answer:\n",
      "\n",
      "task TAP_PNC:\n",
      "Instructions: Answer the following question about antibody developability.\n",
      "Context: Immunogenicity, instability, self-association, high viscosity, polyspecificity, or poor expression can all preclude an antibody from becoming a therapeutic. Early identification of these negative characteristics is essential. Akin to the Lipinski guidelines, which measure druglikeness in small molecules, Therapeutic Antibody Profiler (TAP) highlights antibodies that possess characteristics that are rare/unseen in clinical-stage mAb therapeutics. One key criteria is positive charge and negative charge in the CDRs. Surface patches of positive or negative charge in the CDRs are linked to high rates of clearance and poor expression levels. PNC is a score highlighting regions of dense negative charge.\n",
      "Question: Given an antibody heavy chain and light chain sequence, predict its normalized PNC from 000 to 1000, where 000 is minimum PNC and 1000 is maximum PNC.\n",
      "Antibody heavy chain sequence: {Antibody heavy chain sequence}\n",
      "Antibody light chain sequence: {Antibody light chain sequence}\n",
      "Answer:\n",
      "\n",
      "task TAP_SFvCSP:\n",
      "Instructions: Answer the following question about antibody developability.\n",
      "Context: Immunogenicity, instability, self-association, high viscosity, polyspecificity, or poor expression can all preclude an antibody from becoming a therapeutic. Early identification of these negative characteristics is essential. Akin to the Lipinski guidelines, which measure druglikeness in small molecules, Therapeutic Antibody Profiler (TAP) highlights antibodies that possess characteristics that are rare/unseen in clinical-stage mAb therapeutics. One key criteria is asymmetry in the net heavy- and light-chain surface charges. Asymmetry in the net charge of the heavy- and light-chain variable domains is also correlated with self-association and viscosity at high concentrations. The structural Fv charge symmetry parameter, SFvCSP, only includes residues that are surface-exposed, and not locked in salt bridges, in the evaluation of net charge.\n",
      "Question: Given an antibody heavy chain and light chain sequence, predict its normalized SFvCSP from 000 to 1000, where 000 is minimum SFvCSP and 1000 is maximum SFvCSP.\n",
      "Antibody heavy chain sequence: {Antibody heavy chain sequence}\n",
      "Antibody light chain sequence: {Antibody light chain sequence}\n",
      "Answer:\n",
      "\n",
      "task Leenay_Fraction_Insertions:\n",
      "Instructions: Answer the following question about CRISPR repair outcomes.\n",
      "Context: CRISPR-Cas9 is a gene editing technology that allows targeted deletion or modification of specific regions of the DNA within an organism. This is achieved through designing a guide RNA sequence that binds upstream of the target site which is then cleaved through a Cas9-mediated double stranded DNA break. The cell responds by employing DNA repair mechanisms (such as non-homologous end joining) that result in heterogeneous outcomes including gene insertion or deletion mutations (indels) of varying lengths and frequencies. Primary T cells are a promising cell type for therapeutic genome editing, as they can be engineered efficiently ex vivo and transferred to patients. An experiment is carried out to quantify the DNA repair outcomes of CRISPR-CAS9 knockout experiments on primary CD4+ T cells drawn from donors. For a genomic location, a GuideSeq sequence is provided, which is a 20-nucleotide guide sequence along with the 3-nucletoide PAM  sequence. The PAM (protospacer adjacent motif) is a short DNA sequence that follows the DNA region targeted for cleavage by the CRISPR-Cas9 system. One particular measure of interest for repair outcome quantification is the fraction of indel reads with an insertion.\n",
      "Question: Given a GuideSeq sequence, predict its normalized fraction of indel reads with an insertion from 000 to 1000, where 000 is minimum fraction of indel reads with an insertion and 1000 is maximum fraction of indel reads with an insertion.\n",
      "GuideSeq: {GuideSeq}\n",
      "Answer:\n",
      "\n",
      "task Leenay_Avg_Insertion_Length:\n",
      "Instructions: Answer the following question about CRISPR repair outcomes.\n",
      "Context: CRISPR-Cas9 is a gene editing technology that allows targeted deletion or modification of specific regions of the DNA within an organism. This is achieved through designing a guide RNA sequence that binds upstream of the target site which is then cleaved through a Cas9-mediated double stranded DNA break. The cell responds by employing DNA repair mechanisms (such as non-homologous end joining) that result in heterogeneous outcomes including gene insertion or deletion mutations (indels) of varying lengths and frequencies. Primary T cells are a promising cell type for therapeutic genome editing, as they can be engineered efficiently ex vivo and transferred to patients. An experiment is carried out to quantify the DNA repair outcomes of CRISPR-CAS9 knockout experiments on primary CD4+ T cells drawn from donors. For a genomic location, a GuideSeq sequence is provided, which is a 20-nucleotide guide sequence along with the 3-nucletoide PAM  sequence. The PAM (protospacer adjacent motif) is a short DNA sequence that follows the DNA region targeted for cleavage by the CRISPR-Cas9 system. One particular measure of interest for repair outcome quantification is the average insertion length.\n",
      "Question: Given a GuideSeq sequence, predict its normalized average insertion length from 000 to 1000, where 000 is minimum average insertion length and 1000 is maximum average insertion length.\n",
      "GuideSeq: {GuideSeq}\n",
      "Answer:\n",
      "\n",
      "task Leenay_Avg_Deletion_Length:\n",
      "Instructions: Answer the following question about CRISPR repair outcomes.\n",
      "Context: CRISPR-Cas9 is a gene editing technology that allows targeted deletion or modification of specific regions of the DNA within an organism. This is achieved through designing a guide RNA sequence that binds upstream of the target site which is then cleaved through a Cas9-mediated double stranded DNA break. The cell responds by employing DNA repair mechanisms (such as non-homologous end joining) that result in heterogeneous outcomes including gene insertion or deletion mutations (indels) of varying lengths and frequencies. Primary T cells are a promising cell type for therapeutic genome editing, as they can be engineered efficiently ex vivo and transferred to patients. An experiment is carried out to quantify the DNA repair outcomes of CRISPR-CAS9 knockout experiments on primary CD4+ T cells drawn from donors. For a genomic location, a GuideSeq sequence is provided, which is a 20-nucleotide guide sequence along with the 3-nucletoide PAM  sequence. The PAM (protospacer adjacent motif) is a short DNA sequence that follows the DNA region targeted for cleavage by the CRISPR-Cas9 system. One particular measure of interest for repair outcome quantification is the average deletion length.\n",
      "Question: Given a GuideSeq sequence, predict its normalized average deletion length from 000 to 1000, where 000 is minimum average deletion length and 1000 is maximum average deletion length.\n",
      "GuideSeq: {GuideSeq}\n",
      "Answer:\n",
      "\n",
      "task Leenay_Indel_Diversity:\n",
      "Instructions: Answer the following question about CRISPR repair outcomes.\n",
      "Context: CRISPR-Cas9 is a gene editing technology that allows targeted deletion or modification of specific regions of the DNA within an organism. This is achieved through designing a guide RNA sequence that binds upstream of the target site which is then cleaved through a Cas9-mediated double stranded DNA break. The cell responds by employing DNA repair mechanisms (such as non-homologous end joining) that result in heterogeneous outcomes including gene insertion or deletion mutations (indels) of varying lengths and frequencies. Primary T cells are a promising cell type for therapeutic genome editing, as they can be engineered efficiently ex vivo and transferred to patients. An experiment is carried out to quantify the DNA repair outcomes of CRISPR-CAS9 knockout experiments on primary CD4+ T cells drawn from donors. For a genomic location, a GuideSeq sequence is provided, which is a 20-nucleotide guide sequence along with the 3-nucletoide PAM  sequence. The PAM (protospacer adjacent motif) is a short DNA sequence that follows the DNA region targeted for cleavage by the CRISPR-Cas9 system. One particular measure of interest for repair outcome quantification is the indel diveristy.\n",
      "Question: Given a GuideSeq sequence, predict its normalized indel diversity from 000 to 1000, where 000 is minimum indel diversity and 1000 is maximum indel diversity.\n",
      "GuideSeq: {GuideSeq}\n",
      "Answer:\n",
      "\n",
      "task Leenay_Fraction_Frameshifts:\n",
      "Instructions: Answer the following question about CRISPR repair outcomes.\n",
      "Context: CRISPR-Cas9 is a gene editing technology that allows targeted deletion or modification of specific regions of the DNA within an organism. This is achieved through designing a guide RNA sequence that binds upstream of the target site which is then cleaved through a Cas9-mediated double stranded DNA break. The cell responds by employing DNA repair mechanisms (such as non-homologous end joining) that result in heterogeneous outcomes including gene insertion or deletion mutations (indels) of varying lengths and frequencies. Primary T cells are a promising cell type for therapeutic genome editing, as they can be engineered efficiently ex vivo and transferred to patients. An experiment is carried out to quantify the DNA repair outcomes of CRISPR-CAS9 knockout experiments on primary CD4+ T cells drawn from donors. For a genomic location, a GuideSeq sequence is provided, which is a 20-nucleotide guide sequence along with the 3-nucletoide PAM  sequence. The PAM (protospacer adjacent motif) is a short DNA sequence that follows the DNA region targeted for cleavage by the CRISPR-Cas9 system. One particular measure of interest for repair outcome quantification is the fraction of repair outcomes with a frameshift.\n",
      "Question: Given a GuideSeq sequence, predict its normalized fraction of repair outcomes with a frameshift from 000 to 1000, where 000 is minimum fraction of repair outcomes with a frameshift and 1000 is maximum fraction of repair outcomes with a frameshift.\n",
      "GuideSeq: {GuideSeq}\n",
      "Answer:\n",
      "\n",
      "task KIBA:\n",
      "Instructions: Answer the following question about drug target interactions.\n",
      "Context: Drug-target interaction prediction task aims to predict the interaction activity score in silico given only the accessible compound structural information and protein amino acid sequence.\n",
      "Question: Given the drug SMILES string and target amino acid sequence predict the normalized binding affinity from 000 to 1000, where 000 is minimum binding affinity and 1000 is maximum binding affinity.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Target amino acid sequence: {Target amino acid sequence}\n",
      "Answer:\n",
      "\n",
      "task BindingDB_Patent:\n",
      "Instructions: Answer the following question about drug target interactions.\n",
      "Context: Drug-target interaction prediction task aims to predict the interaction activity score in silico given only the accessible compound structural information and protein amino acid sequence.\n",
      "Question: Given the drug SMILES string and target amino acid sequence predict the normalized binding affinity from 000 to 1000, where 000 is minimum binding affinity and 1000 is maximum binding affinity.\n",
      "Drug SMILES: {Drug SMILES}\n",
      "Target amino acid sequence: {Target amino acid sequence}\n",
      "Answer:\n",
      "\n",
      "task USPTO:\n",
      "Instructions: Answer the following question about reactions.\n",
      "Context: Retrosynthesis is the process of finding a set of reactants that can synthesize a target molecule, i.e., product, which is a fundamental task in drug manufacturing. The target is recursively transformed into simpler precursor molecules until commercially available \"starting\" molecules are identified. In a data sample, there is only one product molecule, reactants can be one or multiple molecules.\n",
      "Question: Given a product SMILES string, predict the reactant SMILES string.\n",
      "\n",
      "Product SMILES: {Product SMILES}\n",
      "Answer:\n",
      "\n"
     ]
    }
   ],
   "execution_count": 14
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-07T22:34:23.179052Z",
     "start_time": "2025-10-07T22:34:23.172526Z"
    }
   },
   "cell_type": "code",
   "source": "list(tdc_prompts_json.keys())",
   "id": "a787ae5f0e0c7183",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Pgp_Broccatelli',\n",
       " 'Bioavailability_Ma',\n",
       " 'BBB_Martins',\n",
       " 'CYP2D6_Veith',\n",
       " 'CYP3A4_Veith',\n",
       " 'CYP2C9_Veith',\n",
       " 'CYP2D6_Substrate_CarbonMangels',\n",
       " 'CYP3A4_Substrate_CarbonMangels',\n",
       " 'CYP2C9_Substrate_CarbonMangels',\n",
       " 'hERG',\n",
       " 'AMES',\n",
       " 'DILI',\n",
       " 'HIA_Hou',\n",
       " 'Tox21_NR_AhR',\n",
       " 'Tox21_NR_Aromatase',\n",
       " 'Tox21_NR_AR',\n",
       " 'Tox21_NR_AR_LBD',\n",
       " 'Tox21_NR_ER',\n",
       " 'Tox21_NR_ER_LBD',\n",
       " 'Tox21_NR_PPAR_gamma',\n",
       " 'Tox21_SR_ARE',\n",
       " 'Tox21_SR_ATAD5',\n",
       " 'Tox21_SR_HSE',\n",
       " 'Tox21_SR_MMP',\n",
       " 'Tox21_SR_p53',\n",
       " 'PAMPA_NCATS',\n",
       " 'CYP2C19_Veith',\n",
       " 'CYP1A2_Veith',\n",
       " 'Skin_Reaction',\n",
       " 'Carcinogens_Lagunin',\n",
       " 'SARSCoV2_Vitro_Touret',\n",
       " 'SARSCOV2_3CLPro_Diamond',\n",
       " 'HIV',\n",
       " 'ClinTox',\n",
       " 'orexin1_receptor_butkiewicz',\n",
       " 'm1_muscarinic_receptor_agonists_butkiewicz',\n",
       " 'm1_muscarinic_receptor_antagonists_butkiewicz',\n",
       " 'potassium_ion_channel_kir2.1_butkiewicz',\n",
       " 'kcnq2_potassium_channel_butkiewicz',\n",
       " 'cav3_t_type_calcium_channels_butkiewicz',\n",
       " 'choline_transporter_butkiewicz',\n",
       " 'serine_threonine_kinase_33_butkiewicz',\n",
       " 'tyrosyl_dna_phosphodiesterase_butkiewicz',\n",
       " 'herg_central',\n",
       " 'hERG_Karim',\n",
       " 'MHC1_IEDB_IMGT_Nielsen',\n",
       " 'MHC2_IEDB_Jensen',\n",
       " 'weber',\n",
       " 'HuRI',\n",
       " 'SAbDab_Chen',\n",
       " 'miRTarBase',\n",
       " 'ToxCast_ACEA_T47D_80hr_Negative',\n",
       " 'ToxCast_ACEA_T47D_80hr_Positive',\n",
       " 'ToxCast_APR_HepG2_CellCycleArrest_24h_dn',\n",
       " 'ToxCast_APR_HepG2_CellCycleArrest_24h_up',\n",
       " 'ToxCast_APR_HepG2_CellCycleArrest_72h_dn',\n",
       " 'ToxCast_APR_HepG2_CellLoss_24h_dn',\n",
       " 'ToxCast_APR_HepG2_CellLoss_72h_dn',\n",
       " 'ToxCast_APR_HepG2_MicrotubuleCSK_24h_dn',\n",
       " 'ToxCast_APR_HepG2_MicrotubuleCSK_24h_up',\n",
       " 'ToxCast_APR_HepG2_MicrotubuleCSK_72h_dn',\n",
       " 'ToxCast_APR_HepG2_MicrotubuleCSK_72h_up',\n",
       " 'ToxCast_APR_HepG2_MitoMass_24h_dn',\n",
       " 'ToxCast_APR_HepG2_MitoMass_24h_up',\n",
       " 'ToxCast_APR_HepG2_MitoMass_72h_dn',\n",
       " 'ToxCast_APR_HepG2_MitoMass_72h_up',\n",
       " 'ToxCast_APR_HepG2_MitoMembPot_1h_dn',\n",
       " 'ToxCast_APR_HepG2_MitoMembPot_24h_dn',\n",
       " 'ToxCast_APR_HepG2_MitoMembPot_72h_dn',\n",
       " 'ToxCast_APR_HepG2_MitoticArrest_24h_up',\n",
       " 'ToxCast_APR_HepG2_MitoticArrest_72h_up',\n",
       " 'ToxCast_APR_HepG2_NuclearSize_24h_dn',\n",
       " 'ToxCast_APR_HepG2_NuclearSize_72h_dn',\n",
       " 'ToxCast_APR_HepG2_NuclearSize_72h_up',\n",
       " 'ToxCast_APR_HepG2_OxidativeStress_24h_up',\n",
       " 'ToxCast_APR_HepG2_OxidativeStress_72h_up',\n",
       " 'ToxCast_APR_HepG2_StressKinase_1h_up',\n",
       " 'ToxCast_APR_HepG2_StressKinase_24h_up',\n",
       " 'ToxCast_APR_HepG2_StressKinase_72h_up',\n",
       " 'ToxCast_APR_HepG2_p53Act_24h_up',\n",
       " 'ToxCast_APR_HepG2_p53Act_72h_up',\n",
       " 'ToxCast_APR_Hepat_Apoptosis_24hr_up',\n",
       " 'ToxCast_APR_Hepat_Apoptosis_48hr_up',\n",
       " 'ToxCast_APR_Hepat_CellLoss_24hr_dn',\n",
       " 'ToxCast_APR_Hepat_CellLoss_48hr_dn',\n",
       " 'ToxCast_APR_Hepat_DNADamage_24hr_up',\n",
       " 'ToxCast_APR_Hepat_DNADamage_48hr_up',\n",
       " 'ToxCast_APR_Hepat_DNATexture_24hr_up',\n",
       " 'ToxCast_APR_Hepat_DNATexture_48hr_up',\n",
       " 'ToxCast_APR_Hepat_MitoFxnI_1hr_dn',\n",
       " 'ToxCast_APR_Hepat_MitoFxnI_24hr_dn',\n",
       " 'ToxCast_APR_Hepat_MitoFxnI_48hr_dn',\n",
       " 'ToxCast_APR_Hepat_NuclearSize_24hr_dn',\n",
       " 'ToxCast_APR_Hepat_NuclearSize_48hr_dn',\n",
       " 'ToxCast_APR_Hepat_Steatosis_24hr_up',\n",
       " 'ToxCast_APR_Hepat_Steatosis_48hr_up',\n",
       " 'ToxCast_ATG_AP_1_CIS_dn',\n",
       " 'ToxCast_ATG_AP_1_CIS_up',\n",
       " 'ToxCast_ATG_AP_2_CIS_dn',\n",
       " 'ToxCast_ATG_AP_2_CIS_up',\n",
       " 'ToxCast_ATG_AR_TRANS_dn',\n",
       " 'ToxCast_ATG_AR_TRANS_up',\n",
       " 'ToxCast_ATG_Ahr_CIS_dn',\n",
       " 'ToxCast_ATG_Ahr_CIS_up',\n",
       " 'ToxCast_ATG_BRE_CIS_dn',\n",
       " 'ToxCast_ATG_BRE_CIS_up',\n",
       " 'ToxCast_ATG_CAR_TRANS_dn',\n",
       " 'ToxCast_ATG_CAR_TRANS_up',\n",
       " 'ToxCast_ATG_CMV_CIS_dn',\n",
       " 'ToxCast_ATG_CMV_CIS_up',\n",
       " 'ToxCast_ATG_CRE_CIS_dn',\n",
       " 'ToxCast_ATG_CRE_CIS_up',\n",
       " 'ToxCast_ATG_C_EBP_CIS_dn',\n",
       " 'ToxCast_ATG_C_EBP_CIS_up',\n",
       " 'ToxCast_ATG_DR4_LXR_CIS_dn',\n",
       " 'ToxCast_ATG_DR4_LXR_CIS_up',\n",
       " 'ToxCast_ATG_DR5_CIS_dn',\n",
       " 'ToxCast_ATG_DR5_CIS_up',\n",
       " 'ToxCast_ATG_E2F_CIS_dn',\n",
       " 'ToxCast_ATG_E2F_CIS_up',\n",
       " 'ToxCast_ATG_EGR_CIS_up',\n",
       " 'ToxCast_ATG_ERE_CIS_dn',\n",
       " 'ToxCast_ATG_ERE_CIS_up',\n",
       " 'ToxCast_ATG_ERRa_TRANS_dn',\n",
       " 'ToxCast_ATG_ERRg_TRANS_dn',\n",
       " 'ToxCast_ATG_ERRg_TRANS_up',\n",
       " 'ToxCast_ATG_ERa_TRANS_up',\n",
       " 'ToxCast_ATG_E_Box_CIS_dn',\n",
       " 'ToxCast_ATG_E_Box_CIS_up',\n",
       " 'ToxCast_ATG_Ets_CIS_dn',\n",
       " 'ToxCast_ATG_Ets_CIS_up',\n",
       " 'ToxCast_ATG_FXR_TRANS_up',\n",
       " 'ToxCast_ATG_FoxA2_CIS_dn',\n",
       " 'ToxCast_ATG_FoxA2_CIS_up',\n",
       " 'ToxCast_ATG_FoxO_CIS_dn',\n",
       " 'ToxCast_ATG_FoxO_CIS_up',\n",
       " 'ToxCast_ATG_GAL4_TRANS_dn',\n",
       " 'ToxCast_ATG_GATA_CIS_dn',\n",
       " 'ToxCast_ATG_GATA_CIS_up',\n",
       " 'ToxCast_ATG_GLI_CIS_dn',\n",
       " 'ToxCast_ATG_GLI_CIS_up',\n",
       " 'ToxCast_ATG_GRE_CIS_dn',\n",
       " 'ToxCast_ATG_GRE_CIS_up',\n",
       " 'ToxCast_ATG_GR_TRANS_dn',\n",
       " 'ToxCast_ATG_GR_TRANS_up',\n",
       " 'ToxCast_ATG_HIF1a_CIS_dn',\n",
       " 'ToxCast_ATG_HIF1a_CIS_up',\n",
       " 'ToxCast_ATG_HNF4a_TRANS_dn',\n",
       " 'ToxCast_ATG_HNF4a_TRANS_up',\n",
       " 'ToxCast_ATG_HNF6_CIS_dn',\n",
       " 'ToxCast_ATG_HNF6_CIS_up',\n",
       " 'ToxCast_ATG_HSE_CIS_dn',\n",
       " 'ToxCast_ATG_HSE_CIS_up',\n",
       " 'ToxCast_ATG_IR1_CIS_dn',\n",
       " 'ToxCast_ATG_IR1_CIS_up',\n",
       " 'ToxCast_ATG_ISRE_CIS_dn',\n",
       " 'ToxCast_ATG_ISRE_CIS_up',\n",
       " 'ToxCast_ATG_LXRa_TRANS_dn',\n",
       " 'ToxCast_ATG_LXRa_TRANS_up',\n",
       " 'ToxCast_ATG_LXRb_TRANS_dn',\n",
       " 'ToxCast_ATG_LXRb_TRANS_up',\n",
       " 'ToxCast_ATG_MRE_CIS_up',\n",
       " 'ToxCast_ATG_M_06_TRANS_up',\n",
       " 'ToxCast_ATG_M_19_CIS_dn',\n",
       " 'ToxCast_ATG_M_19_TRANS_dn',\n",
       " 'ToxCast_ATG_M_19_TRANS_up',\n",
       " 'ToxCast_ATG_M_32_CIS_dn',\n",
       " 'ToxCast_ATG_M_32_CIS_up',\n",
       " 'ToxCast_ATG_M_32_TRANS_dn',\n",
       " 'ToxCast_ATG_M_32_TRANS_up',\n",
       " 'ToxCast_ATG_M_61_TRANS_up',\n",
       " 'ToxCast_ATG_Myb_CIS_dn',\n",
       " 'ToxCast_ATG_Myb_CIS_up',\n",
       " 'ToxCast_ATG_Myc_CIS_dn',\n",
       " 'ToxCast_ATG_Myc_CIS_up',\n",
       " 'ToxCast_ATG_NFI_CIS_dn',\n",
       " 'ToxCast_ATG_NFI_CIS_up',\n",
       " 'ToxCast_ATG_NF_kB_CIS_dn',\n",
       " 'ToxCast_ATG_NF_kB_CIS_up',\n",
       " 'ToxCast_ATG_NRF1_CIS_dn',\n",
       " 'ToxCast_ATG_NRF1_CIS_up',\n",
       " 'ToxCast_ATG_NRF2_ARE_CIS_dn',\n",
       " 'ToxCast_ATG_NRF2_ARE_CIS_up',\n",
       " 'ToxCast_ATG_NURR1_TRANS_dn',\n",
       " 'ToxCast_ATG_NURR1_TRANS_up',\n",
       " 'ToxCast_ATG_Oct_MLP_CIS_dn',\n",
       " 'ToxCast_ATG_Oct_MLP_CIS_up',\n",
       " 'ToxCast_ATG_PBREM_CIS_dn',\n",
       " 'ToxCast_ATG_PBREM_CIS_up',\n",
       " 'ToxCast_ATG_PPARa_TRANS_dn',\n",
       " 'ToxCast_ATG_PPARa_TRANS_up',\n",
       " 'ToxCast_ATG_PPARd_TRANS_up',\n",
       " 'ToxCast_ATG_PPARg_TRANS_up',\n",
       " 'ToxCast_ATG_PPRE_CIS_dn',\n",
       " 'ToxCast_ATG_PPRE_CIS_up',\n",
       " 'ToxCast_ATG_PXRE_CIS_dn',\n",
       " 'ToxCast_ATG_PXRE_CIS_up',\n",
       " 'ToxCast_ATG_PXR_TRANS_dn',\n",
       " 'ToxCast_ATG_PXR_TRANS_up',\n",
       " 'ToxCast_ATG_Pax6_CIS_up',\n",
       " 'ToxCast_ATG_RARa_TRANS_dn',\n",
       " 'ToxCast_ATG_RARa_TRANS_up',\n",
       " 'ToxCast_ATG_RARb_TRANS_dn',\n",
       " 'ToxCast_ATG_RARb_TRANS_up',\n",
       " 'ToxCast_ATG_RARg_TRANS_dn',\n",
       " 'ToxCast_ATG_RARg_TRANS_up',\n",
       " 'ToxCast_ATG_RORE_CIS_dn',\n",
       " 'ToxCast_ATG_RORE_CIS_up',\n",
       " 'ToxCast_ATG_RORb_TRANS_dn',\n",
       " 'ToxCast_ATG_RORg_TRANS_dn',\n",
       " 'ToxCast_ATG_RORg_TRANS_up',\n",
       " 'ToxCast_ATG_RXRa_TRANS_dn',\n",
       " 'ToxCast_ATG_RXRa_TRANS_up',\n",
       " 'ToxCast_ATG_RXRb_TRANS_dn',\n",
       " 'ToxCast_ATG_RXRb_TRANS_up',\n",
       " 'ToxCast_ATG_SREBP_CIS_dn',\n",
       " 'ToxCast_ATG_SREBP_CIS_up',\n",
       " 'ToxCast_ATG_STAT3_CIS_dn',\n",
       " 'ToxCast_ATG_STAT3_CIS_up',\n",
       " 'ToxCast_ATG_Sox_CIS_dn',\n",
       " 'ToxCast_ATG_Sox_CIS_up',\n",
       " 'ToxCast_ATG_Sp1_CIS_dn',\n",
       " 'ToxCast_ATG_Sp1_CIS_up',\n",
       " 'ToxCast_ATG_TAL_CIS_dn',\n",
       " 'ToxCast_ATG_TAL_CIS_up',\n",
       " 'ToxCast_ATG_TA_CIS_dn',\n",
       " 'ToxCast_ATG_TA_CIS_up',\n",
       " 'ToxCast_ATG_TCF_b_cat_CIS_dn',\n",
       " 'ToxCast_ATG_TCF_b_cat_CIS_up',\n",
       " 'ToxCast_ATG_TGFb_CIS_dn',\n",
       " 'ToxCast_ATG_TGFb_CIS_up',\n",
       " 'ToxCast_ATG_THRa1_TRANS_dn',\n",
       " 'ToxCast_ATG_THRa1_TRANS_up',\n",
       " 'ToxCast_ATG_VDRE_CIS_dn',\n",
       " 'ToxCast_ATG_VDRE_CIS_up',\n",
       " 'ToxCast_ATG_VDR_TRANS_dn',\n",
       " 'ToxCast_ATG_VDR_TRANS_up',\n",
       " 'ToxCast_ATG_XTT_Cytotoxicity_up',\n",
       " 'ToxCast_ATG_Xbp1_CIS_dn',\n",
       " 'ToxCast_ATG_Xbp1_CIS_up',\n",
       " 'ToxCast_ATG_p53_CIS_dn',\n",
       " 'ToxCast_ATG_p53_CIS_up',\n",
       " 'ToxCast_BSK_3C_Eselectin_down',\n",
       " 'ToxCast_BSK_3C_HLADR_down',\n",
       " 'ToxCast_BSK_3C_ICAM1_down',\n",
       " 'ToxCast_BSK_3C_IL8_down',\n",
       " 'ToxCast_BSK_3C_MCP1_down',\n",
       " 'ToxCast_BSK_3C_MIG_down',\n",
       " 'ToxCast_BSK_3C_Proliferation_down',\n",
       " 'ToxCast_BSK_3C_SRB_down',\n",
       " 'ToxCast_BSK_3C_Thrombomodulin_down',\n",
       " 'ToxCast_BSK_3C_Thrombomodulin_up',\n",
       " 'ToxCast_BSK_3C_TissueFactor_down',\n",
       " 'ToxCast_BSK_3C_TissueFactor_up',\n",
       " 'ToxCast_BSK_3C_VCAM1_down',\n",
       " 'ToxCast_BSK_3C_Vis_down',\n",
       " 'ToxCast_BSK_3C_uPAR_down',\n",
       " 'ToxCast_BSK_4H_Eotaxin3_down',\n",
       " 'ToxCast_BSK_4H_MCP1_down',\n",
       " 'ToxCast_BSK_4H_Pselectin_down',\n",
       " 'ToxCast_BSK_4H_Pselectin_up',\n",
       " 'ToxCast_BSK_4H_SRB_down',\n",
       " 'ToxCast_BSK_4H_VCAM1_down',\n",
       " 'ToxCast_BSK_4H_VEGFRII_down',\n",
       " 'ToxCast_BSK_4H_uPAR_down',\n",
       " 'ToxCast_BSK_4H_uPAR_up',\n",
       " 'ToxCast_BSK_BE3C_HLADR_down',\n",
       " 'ToxCast_BSK_BE3C_IL1a_down',\n",
       " 'ToxCast_BSK_BE3C_IP10_down',\n",
       " 'ToxCast_BSK_BE3C_MIG_down',\n",
       " 'ToxCast_BSK_BE3C_MMP1_down',\n",
       " 'ToxCast_BSK_BE3C_MMP1_up',\n",
       " 'ToxCast_BSK_BE3C_PAI1_down',\n",
       " 'ToxCast_BSK_BE3C_SRB_down',\n",
       " 'ToxCast_BSK_BE3C_TGFb1_down',\n",
       " 'ToxCast_BSK_BE3C_tPA_down',\n",
       " 'ToxCast_BSK_BE3C_uPAR_down',\n",
       " 'ToxCast_BSK_BE3C_uPAR_up',\n",
       " 'ToxCast_BSK_BE3C_uPA_down',\n",
       " 'ToxCast_BSK_CASM3C_HLADR_down',\n",
       " 'ToxCast_BSK_CASM3C_IL6_down',\n",
       " 'ToxCast_BSK_CASM3C_IL6_up',\n",
       " 'ToxCast_BSK_CASM3C_IL8_down',\n",
       " 'ToxCast_BSK_CASM3C_LDLR_down',\n",
       " 'ToxCast_BSK_CASM3C_LDLR_up',\n",
       " 'ToxCast_BSK_CASM3C_MCP1_down',\n",
       " 'ToxCast_BSK_CASM3C_MCP1_up',\n",
       " 'ToxCast_BSK_CASM3C_MCSF_down',\n",
       " 'ToxCast_BSK_CASM3C_MCSF_up',\n",
       " 'ToxCast_BSK_CASM3C_MIG_down',\n",
       " 'ToxCast_BSK_CASM3C_Proliferation_down',\n",
       " 'ToxCast_BSK_CASM3C_Proliferation_up',\n",
       " 'ToxCast_BSK_CASM3C_SAA_down',\n",
       " 'ToxCast_BSK_CASM3C_SAA_up',\n",
       " 'ToxCast_BSK_CASM3C_SRB_down',\n",
       " 'ToxCast_BSK_CASM3C_Thrombomodulin_down',\n",
       " 'ToxCast_BSK_CASM3C_Thrombomodulin_up',\n",
       " 'ToxCast_BSK_CASM3C_TissueFactor_down',\n",
       " 'ToxCast_BSK_CASM3C_VCAM1_down',\n",
       " 'ToxCast_BSK_CASM3C_VCAM1_up',\n",
       " 'ToxCast_BSK_CASM3C_uPAR_down',\n",
       " 'ToxCast_BSK_CASM3C_uPAR_up',\n",
       " 'ToxCast_BSK_KF3CT_ICAM1_down',\n",
       " 'ToxCast_BSK_KF3CT_IL1a_down',\n",
       " 'ToxCast_BSK_KF3CT_IP10_down',\n",
       " 'ToxCast_BSK_KF3CT_IP10_up',\n",
       " 'ToxCast_BSK_KF3CT_MCP1_down',\n",
       " 'ToxCast_BSK_KF3CT_MCP1_up',\n",
       " 'ToxCast_BSK_KF3CT_MMP9_down',\n",
       " 'ToxCast_BSK_KF3CT_SRB_down',\n",
       " 'ToxCast_BSK_KF3CT_TGFb1_down',\n",
       " 'ToxCast_BSK_KF3CT_TIMP2_down',\n",
       " 'ToxCast_BSK_KF3CT_uPA_down',\n",
       " 'ToxCast_BSK_LPS_CD40_down',\n",
       " 'ToxCast_BSK_LPS_Eselectin_down',\n",
       " 'ToxCast_BSK_LPS_Eselectin_up',\n",
       " 'ToxCast_BSK_LPS_IL1a_down',\n",
       " 'ToxCast_BSK_LPS_IL1a_up',\n",
       " 'ToxCast_BSK_LPS_IL8_down',\n",
       " 'ToxCast_BSK_LPS_IL8_up',\n",
       " 'ToxCast_BSK_LPS_MCP1_down',\n",
       " 'ToxCast_BSK_LPS_MCSF_down',\n",
       " 'ToxCast_BSK_LPS_PGE2_down',\n",
       " 'ToxCast_BSK_LPS_PGE2_up',\n",
       " 'ToxCast_BSK_LPS_SRB_down',\n",
       " 'ToxCast_BSK_LPS_TNFa_down',\n",
       " 'ToxCast_BSK_LPS_TNFa_up',\n",
       " 'ToxCast_BSK_LPS_TissueFactor_down',\n",
       " 'ToxCast_BSK_LPS_TissueFactor_up',\n",
       " 'ToxCast_BSK_LPS_VCAM1_down',\n",
       " 'ToxCast_BSK_SAg_CD38_down',\n",
       " 'ToxCast_BSK_SAg_CD40_down',\n",
       " 'ToxCast_BSK_SAg_CD69_down',\n",
       " 'ToxCast_BSK_SAg_Eselectin_down',\n",
       " 'ToxCast_BSK_SAg_Eselectin_up',\n",
       " 'ToxCast_BSK_SAg_IL8_down',\n",
       " 'ToxCast_BSK_SAg_IL8_up',\n",
       " 'ToxCast_BSK_SAg_MCP1_down',\n",
       " 'ToxCast_BSK_SAg_MIG_down',\n",
       " 'ToxCast_BSK_SAg_PBMCCytotoxicity_down',\n",
       " 'ToxCast_BSK_SAg_PBMCCytotoxicity_up',\n",
       " 'ToxCast_BSK_SAg_Proliferation_down',\n",
       " 'ToxCast_BSK_SAg_SRB_down',\n",
       " 'ToxCast_BSK_hDFCGF_CollagenIII_down',\n",
       " 'ToxCast_BSK_hDFCGF_EGFR_down',\n",
       " 'ToxCast_BSK_hDFCGF_EGFR_up',\n",
       " 'ToxCast_BSK_hDFCGF_IL8_down',\n",
       " 'ToxCast_BSK_hDFCGF_IP10_down',\n",
       " 'ToxCast_BSK_hDFCGF_MCSF_down',\n",
       " 'ToxCast_BSK_hDFCGF_MIG_down',\n",
       " 'ToxCast_BSK_hDFCGF_MMP1_down',\n",
       " 'ToxCast_BSK_hDFCGF_MMP1_up',\n",
       " 'ToxCast_BSK_hDFCGF_PAI1_down',\n",
       " 'ToxCast_BSK_hDFCGF_Proliferation_down',\n",
       " 'ToxCast_BSK_hDFCGF_SRB_down',\n",
       " 'ToxCast_BSK_hDFCGF_TIMP1_down',\n",
       " 'ToxCast_BSK_hDFCGF_VCAM1_down',\n",
       " 'ToxCast_CEETOX_H295R_11DCORT_dn',\n",
       " 'ToxCast_CEETOX_H295R_ANDR_dn',\n",
       " 'ToxCast_CEETOX_H295R_CORTISOL_dn',\n",
       " 'ToxCast_CEETOX_H295R_DOC_dn',\n",
       " 'ToxCast_CEETOX_H295R_DOC_up',\n",
       " 'ToxCast_CEETOX_H295R_ESTRADIOL_dn',\n",
       " 'ToxCast_CEETOX_H295R_ESTRADIOL_up',\n",
       " 'ToxCast_CEETOX_H295R_ESTRONE_dn',\n",
       " 'ToxCast_CEETOX_H295R_ESTRONE_up',\n",
       " 'ToxCast_CEETOX_H295R_OHPREG_up',\n",
       " 'ToxCast_CEETOX_H295R_OHPROG_dn',\n",
       " 'ToxCast_CEETOX_H295R_OHPROG_up',\n",
       " 'ToxCast_CEETOX_H295R_PROG_up',\n",
       " 'ToxCast_CEETOX_H295R_TESTO_dn',\n",
       " 'ToxCast_CLD_ABCB1_48hr',\n",
       " 'ToxCast_CLD_ABCG2_48hr',\n",
       " 'ToxCast_CLD_CYP1A1_24hr',\n",
       " 'ToxCast_CLD_CYP1A1_48hr',\n",
       " 'ToxCast_CLD_CYP1A1_6hr',\n",
       " 'ToxCast_CLD_CYP1A2_24hr',\n",
       " 'ToxCast_CLD_CYP1A2_48hr',\n",
       " 'ToxCast_CLD_CYP1A2_6hr',\n",
       " 'ToxCast_CLD_CYP2B6_24hr',\n",
       " 'ToxCast_CLD_CYP2B6_48hr',\n",
       " 'ToxCast_CLD_CYP2B6_6hr',\n",
       " 'ToxCast_CLD_CYP3A4_24hr',\n",
       " 'ToxCast_CLD_CYP3A4_48hr',\n",
       " 'ToxCast_CLD_CYP3A4_6hr',\n",
       " 'ToxCast_CLD_GSTA2_48hr',\n",
       " 'ToxCast_CLD_SULT2A_24hr',\n",
       " 'ToxCast_CLD_SULT2A_48hr',\n",
       " 'ToxCast_CLD_UGT1A1_24hr',\n",
       " 'ToxCast_CLD_UGT1A1_48hr',\n",
       " 'ToxCast_NCCT_HEK293T_CellTiterGLO',\n",
       " 'ToxCast_NCCT_QuantiLum_inhib_2_dn',\n",
       " 'ToxCast_NCCT_QuantiLum_inhib_dn',\n",
       " 'ToxCast_NCCT_TPO_AUR_dn',\n",
       " 'ToxCast_NCCT_TPO_GUA_dn',\n",
       " 'ToxCast_NHEERL_ZF_144hpf_TERATOSCORE_up',\n",
       " 'ToxCast_NVS_ADME_hCYP19A1',\n",
       " 'ToxCast_NVS_ADME_hCYP1A1',\n",
       " 'ToxCast_NVS_ADME_hCYP1A2',\n",
       " 'ToxCast_NVS_ADME_hCYP2A6',\n",
       " 'ToxCast_NVS_ADME_hCYP2B6',\n",
       " 'ToxCast_NVS_ADME_hCYP2C19',\n",
       " 'ToxCast_NVS_ADME_hCYP2C9',\n",
       " 'ToxCast_NVS_ADME_hCYP2D6',\n",
       " 'ToxCast_NVS_ADME_hCYP3A4',\n",
       " 'ToxCast_NVS_ADME_hCYP4F12',\n",
       " 'ToxCast_NVS_ADME_rCYP2C12',\n",
       " 'ToxCast_NVS_ENZ_hAChE',\n",
       " 'ToxCast_NVS_ENZ_hAMPKa1',\n",
       " 'ToxCast_NVS_ENZ_hAurA',\n",
       " 'ToxCast_NVS_ENZ_hBACE',\n",
       " 'ToxCast_NVS_ENZ_hCASP5',\n",
       " 'ToxCast_NVS_ENZ_hCK1D',\n",
       " 'ToxCast_NVS_ENZ_hDUSP3',\n",
       " 'ToxCast_NVS_ENZ_hES',\n",
       " 'ToxCast_NVS_ENZ_hElastase',\n",
       " 'ToxCast_NVS_ENZ_hFGFR1',\n",
       " 'ToxCast_NVS_ENZ_hGSK3b',\n",
       " 'ToxCast_NVS_ENZ_hMMP1',\n",
       " 'ToxCast_NVS_ENZ_hMMP13',\n",
       " 'ToxCast_NVS_ENZ_hMMP2',\n",
       " 'ToxCast_NVS_ENZ_hMMP3',\n",
       " 'ToxCast_NVS_ENZ_hMMP7',\n",
       " 'ToxCast_NVS_ENZ_hMMP9',\n",
       " 'ToxCast_NVS_ENZ_hPDE10',\n",
       " 'ToxCast_NVS_ENZ_hPDE4A1',\n",
       " 'ToxCast_NVS_ENZ_hPDE5',\n",
       " 'ToxCast_NVS_ENZ_hPI3Ka',\n",
       " 'ToxCast_NVS_ENZ_hPTEN',\n",
       " 'ToxCast_NVS_ENZ_hPTPN11',\n",
       " 'ToxCast_NVS_ENZ_hPTPN12',\n",
       " 'ToxCast_NVS_ENZ_hPTPN13',\n",
       " 'ToxCast_NVS_ENZ_hPTPN9',\n",
       " 'ToxCast_NVS_ENZ_hPTPRC',\n",
       " 'ToxCast_NVS_ENZ_hSIRT1',\n",
       " 'ToxCast_NVS_ENZ_hSIRT2',\n",
       " 'ToxCast_NVS_ENZ_hTrkA',\n",
       " 'ToxCast_NVS_ENZ_hVEGFR2',\n",
       " 'ToxCast_NVS_ENZ_oCOX1',\n",
       " 'ToxCast_NVS_ENZ_oCOX2',\n",
       " 'ToxCast_NVS_ENZ_rAChE',\n",
       " 'ToxCast_NVS_ENZ_rCNOS',\n",
       " 'ToxCast_NVS_ENZ_rMAOAC',\n",
       " 'ToxCast_NVS_ENZ_rMAOAP',\n",
       " 'ToxCast_NVS_ENZ_rMAOBC',\n",
       " 'ToxCast_NVS_ENZ_rMAOBP',\n",
       " 'ToxCast_NVS_ENZ_rabI2C',\n",
       " 'ToxCast_NVS_GPCR_bAdoR_NonSelective',\n",
       " 'ToxCast_NVS_GPCR_bDR_NonSelective',\n",
       " 'ToxCast_NVS_GPCR_g5HT4',\n",
       " 'ToxCast_NVS_GPCR_gH2',\n",
       " 'ToxCast_NVS_GPCR_gLTB4',\n",
       " 'ToxCast_NVS_GPCR_gLTD4',\n",
       " 'ToxCast_NVS_GPCR_gMPeripheral_NonSelective',\n",
       " 'ToxCast_NVS_GPCR_gOpiateK',\n",
       " 'ToxCast_NVS_GPCR_h5HT2A',\n",
       " 'ToxCast_NVS_GPCR_h5HT5A',\n",
       " 'ToxCast_NVS_GPCR_h5HT6',\n",
       " 'ToxCast_NVS_GPCR_h5HT7',\n",
       " 'ToxCast_NVS_GPCR_hAT1',\n",
       " 'ToxCast_NVS_GPCR_hAdoRA1',\n",
       " 'ToxCast_NVS_GPCR_hAdoRA2a',\n",
       " 'ToxCast_NVS_GPCR_hAdra2A',\n",
       " 'ToxCast_NVS_GPCR_hAdra2C',\n",
       " 'ToxCast_NVS_GPCR_hAdrb1',\n",
       " 'ToxCast_NVS_GPCR_hAdrb2',\n",
       " 'ToxCast_NVS_GPCR_hAdrb3',\n",
       " 'ToxCast_NVS_GPCR_hDRD1',\n",
       " 'ToxCast_NVS_GPCR_hDRD2s',\n",
       " 'ToxCast_NVS_GPCR_hDRD4.4',\n",
       " 'ToxCast_NVS_GPCR_hH1',\n",
       " 'ToxCast_NVS_GPCR_hLTB4_BLT1',\n",
       " 'ToxCast_NVS_GPCR_hM1',\n",
       " 'ToxCast_NVS_GPCR_hM2',\n",
       " 'ToxCast_NVS_GPCR_hM3',\n",
       " 'ToxCast_NVS_GPCR_hM4',\n",
       " 'ToxCast_NVS_GPCR_hNK2',\n",
       " 'ToxCast_NVS_GPCR_hOpiate_D1',\n",
       " 'ToxCast_NVS_GPCR_hOpiate_mu',\n",
       " 'ToxCast_NVS_GPCR_hTXA2',\n",
       " 'ToxCast_NVS_GPCR_p5HT2C',\n",
       " 'ToxCast_NVS_GPCR_r5HT1_NonSelective',\n",
       " 'ToxCast_NVS_GPCR_r5HT_NonSelective',\n",
       " 'ToxCast_NVS_GPCR_rAdra1B',\n",
       " 'ToxCast_NVS_GPCR_rAdra1_NonSelective',\n",
       " 'ToxCast_NVS_GPCR_rAdra2_NonSelective',\n",
       " 'ToxCast_NVS_GPCR_rAdrb_NonSelective',\n",
       " 'ToxCast_NVS_GPCR_rNK1',\n",
       " 'ToxCast_NVS_GPCR_rNK3',\n",
       " 'ToxCast_NVS_GPCR_rOpiate_NonSelective',\n",
       " 'ToxCast_NVS_GPCR_rOpiate_NonSelectiveNa',\n",
       " 'ToxCast_NVS_GPCR_rSST',\n",
       " 'ToxCast_NVS_GPCR_rTRH',\n",
       " 'ToxCast_NVS_GPCR_rV1',\n",
       " 'ToxCast_NVS_GPCR_rabPAF',\n",
       " 'ToxCast_NVS_GPCR_rmAdra2B',\n",
       " 'ToxCast_NVS_IC_hKhERGCh',\n",
       " 'ToxCast_NVS_IC_rCaBTZCHL',\n",
       " 'ToxCast_NVS_IC_rCaDHPRCh_L',\n",
       " 'ToxCast_NVS_IC_rNaCh_site2',\n",
       " 'ToxCast_NVS_LGIC_bGABARa1',\n",
       " 'ToxCast_NVS_LGIC_h5HT3',\n",
       " 'ToxCast_NVS_LGIC_hNNR_NBungSens',\n",
       " 'ToxCast_NVS_LGIC_rGABAR_NonSelective',\n",
       " 'ToxCast_NVS_LGIC_rNNR_BungSens',\n",
       " 'ToxCast_NVS_MP_hPBR',\n",
       " 'ToxCast_NVS_MP_rPBR',\n",
       " 'ToxCast_NVS_NR_bER',\n",
       " 'ToxCast_NVS_NR_bPR',\n",
       " 'ToxCast_NVS_NR_cAR',\n",
       " 'ToxCast_NVS_NR_hAR',\n",
       " 'ToxCast_NVS_NR_hCAR_Antagonist',\n",
       " 'ToxCast_NVS_NR_hER',\n",
       " 'ToxCast_NVS_NR_hFXR_Agonist',\n",
       " 'ToxCast_NVS_NR_hFXR_Antagonist',\n",
       " 'ToxCast_NVS_NR_hGR',\n",
       " 'ToxCast_NVS_NR_hPPARa',\n",
       " 'ToxCast_NVS_NR_hPPARg',\n",
       " 'ToxCast_NVS_NR_hPR',\n",
       " 'ToxCast_NVS_NR_hPXR',\n",
       " 'ToxCast_NVS_NR_hRAR_Antagonist',\n",
       " 'ToxCast_NVS_NR_hRARa_Agonist',\n",
       " 'ToxCast_NVS_NR_hTRa_Antagonist',\n",
       " 'ToxCast_NVS_NR_mERa',\n",
       " 'ToxCast_NVS_NR_rAR',\n",
       " 'ToxCast_NVS_NR_rMR',\n",
       " 'ToxCast_NVS_OR_gSIGMA_NonSelective',\n",
       " 'ToxCast_NVS_TR_gDAT',\n",
       " 'ToxCast_NVS_TR_hAdoT',\n",
       " 'ToxCast_NVS_TR_hDAT',\n",
       " 'ToxCast_NVS_TR_hNET',\n",
       " 'ToxCast_NVS_TR_hSERT',\n",
       " 'ToxCast_NVS_TR_rNET',\n",
       " 'ToxCast_NVS_TR_rSERT',\n",
       " 'ToxCast_NVS_TR_rVMAT2',\n",
       " 'ToxCast_OT_AR_ARELUC_AG_1440',\n",
       " 'ToxCast_OT_AR_ARSRC1_0480',\n",
       " 'ToxCast_OT_AR_ARSRC1_0960',\n",
       " 'ToxCast_OT_ER_ERaERa_0480',\n",
       " 'ToxCast_OT_ER_ERaERa_1440',\n",
       " 'ToxCast_OT_ER_ERaERb_0480',\n",
       " 'ToxCast_OT_ER_ERaERb_1440',\n",
       " 'ToxCast_OT_ER_ERbERb_0480',\n",
       " 'ToxCast_OT_ER_ERbERb_1440',\n",
       " 'ToxCast_OT_ERa_EREGFP_0120',\n",
       " 'ToxCast_OT_ERa_EREGFP_0480',\n",
       " 'ToxCast_OT_FXR_FXRSRC1_0480',\n",
       " 'ToxCast_OT_FXR_FXRSRC1_1440',\n",
       " 'ToxCast_OT_NURR1_NURR1RXRa_0480',\n",
       " 'ToxCast_OT_NURR1_NURR1RXRa_1440',\n",
       " 'ToxCast_TOX21_ARE_BLA_Agonist_ch1',\n",
       " 'ToxCast_TOX21_ARE_BLA_Agonist_ch2',\n",
       " 'ToxCast_TOX21_ARE_BLA_agonist_ratio',\n",
       " 'ToxCast_TOX21_ARE_BLA_agonist_viability',\n",
       " 'ToxCast_TOX21_AR_BLA_Agonist_ch1',\n",
       " 'ToxCast_TOX21_AR_BLA_Agonist_ch2',\n",
       " 'ToxCast_TOX21_AR_BLA_Agonist_ratio',\n",
       " 'ToxCast_TOX21_AR_BLA_Antagonist_ch1',\n",
       " 'ToxCast_TOX21_AR_BLA_Antagonist_ch2',\n",
       " 'ToxCast_TOX21_AR_BLA_Antagonist_ratio',\n",
       " 'ToxCast_TOX21_AR_BLA_Antagonist_viability',\n",
       " 'ToxCast_TOX21_AR_LUC_MDAKB2_Agonist',\n",
       " 'ToxCast_TOX21_AR_LUC_MDAKB2_Antagonist',\n",
       " 'ToxCast_TOX21_AR_LUC_MDAKB2_Antagonist2',\n",
       " 'ToxCast_TOX21_AhR_LUC_Agonist',\n",
       " 'ToxCast_TOX21_Aromatase_Inhibition',\n",
       " 'ToxCast_TOX21_AutoFluor_HEK293_Cell_blue',\n",
       " 'ToxCast_TOX21_AutoFluor_HEK293_Media_blue',\n",
       " 'ToxCast_TOX21_AutoFluor_HEPG2_Cell_blue',\n",
       " 'ToxCast_TOX21_AutoFluor_HEPG2_Cell_green',\n",
       " 'ToxCast_TOX21_AutoFluor_HEPG2_Media_blue',\n",
       " 'ToxCast_TOX21_AutoFluor_HEPG2_Media_green',\n",
       " 'ToxCast_TOX21_ELG1_LUC_Agonist',\n",
       " 'ToxCast_TOX21_ERa_BLA_Agonist_ch1',\n",
       " 'ToxCast_TOX21_ERa_BLA_Agonist_ch2',\n",
       " 'ToxCast_TOX21_ERa_BLA_Agonist_ratio',\n",
       " 'ToxCast_TOX21_ERa_BLA_Antagonist_ch1',\n",
       " 'ToxCast_TOX21_ERa_BLA_Antagonist_ch2',\n",
       " 'ToxCast_TOX21_ERa_BLA_Antagonist_ratio',\n",
       " 'ToxCast_TOX21_ERa_BLA_Antagonist_viability',\n",
       " 'ToxCast_TOX21_ERa_LUC_BG1_Agonist',\n",
       " 'ToxCast_TOX21_ERa_LUC_BG1_Antagonist',\n",
       " 'ToxCast_TOX21_ESRE_BLA_ch1',\n",
       " 'ToxCast_TOX21_ESRE_BLA_ch2',\n",
       " 'ToxCast_TOX21_ESRE_BLA_ratio',\n",
       " 'ToxCast_TOX21_ESRE_BLA_viability',\n",
       " 'ToxCast_TOX21_FXR_BLA_Antagonist_ch1',\n",
       " 'ToxCast_TOX21_FXR_BLA_Antagonist_ch2',\n",
       " 'ToxCast_TOX21_FXR_BLA_agonist_ch2',\n",
       " 'ToxCast_TOX21_FXR_BLA_agonist_ratio',\n",
       " 'ToxCast_TOX21_FXR_BLA_antagonist_ratio',\n",
       " 'ToxCast_TOX21_FXR_BLA_antagonist_viability',\n",
       " 'ToxCast_TOX21_GR_BLA_Agonist_ch1',\n",
       " 'ToxCast_TOX21_GR_BLA_Agonist_ch2',\n",
       " 'ToxCast_TOX21_GR_BLA_Agonist_ratio',\n",
       " 'ToxCast_TOX21_GR_BLA_Antagonist_ch2',\n",
       " 'ToxCast_TOX21_GR_BLA_Antagonist_ratio',\n",
       " 'ToxCast_TOX21_GR_BLA_Antagonist_viability',\n",
       " 'ToxCast_TOX21_HSE_BLA_agonist_ch1',\n",
       " 'ToxCast_TOX21_HSE_BLA_agonist_ch2',\n",
       " 'ToxCast_TOX21_HSE_BLA_agonist_ratio',\n",
       " 'ToxCast_TOX21_HSE_BLA_agonist_viability',\n",
       " 'ToxCast_TOX21_MMP_ratio_down',\n",
       " 'ToxCast_TOX21_MMP_ratio_up',\n",
       " 'ToxCast_TOX21_MMP_viability',\n",
       " 'ToxCast_TOX21_NFkB_BLA_agonist_ch1',\n",
       " 'ToxCast_TOX21_NFkB_BLA_agonist_ch2',\n",
       " 'ToxCast_TOX21_NFkB_BLA_agonist_ratio',\n",
       " 'ToxCast_TOX21_NFkB_BLA_agonist_viability',\n",
       " 'ToxCast_TOX21_PPARd_BLA_Agonist_viability',\n",
       " 'ToxCast_TOX21_PPARd_BLA_Antagonist_ch1',\n",
       " 'ToxCast_TOX21_PPARd_BLA_agonist_ch1',\n",
       " 'ToxCast_TOX21_PPARd_BLA_agonist_ch2',\n",
       " 'ToxCast_TOX21_PPARd_BLA_agonist_ratio',\n",
       " 'ToxCast_TOX21_PPARd_BLA_antagonist_ratio',\n",
       " 'ToxCast_TOX21_PPARd_BLA_antagonist_viability',\n",
       " 'ToxCast_TOX21_PPARg_BLA_Agonist_ch1',\n",
       " 'ToxCast_TOX21_PPARg_BLA_Agonist_ch2',\n",
       " 'ToxCast_TOX21_PPARg_BLA_Agonist_ratio',\n",
       " 'ToxCast_TOX21_PPARg_BLA_Antagonist_ch1',\n",
       " 'ToxCast_TOX21_PPARg_BLA_antagonist_ratio',\n",
       " 'ToxCast_TOX21_PPARg_BLA_antagonist_viability',\n",
       " 'ToxCast_TOX21_TR_LUC_GH3_Agonist',\n",
       " 'ToxCast_TOX21_TR_LUC_GH3_Antagonist',\n",
       " 'ToxCast_TOX21_VDR_BLA_Agonist_viability',\n",
       " 'ToxCast_TOX21_VDR_BLA_Antagonist_ch1',\n",
       " 'ToxCast_TOX21_VDR_BLA_agonist_ch2',\n",
       " 'ToxCast_TOX21_VDR_BLA_agonist_ratio',\n",
       " 'ToxCast_TOX21_VDR_BLA_antagonist_ratio',\n",
       " 'ToxCast_TOX21_VDR_BLA_antagonist_viability',\n",
       " 'ToxCast_TOX21_p53_BLA_p1_ch1',\n",
       " 'ToxCast_TOX21_p53_BLA_p1_ch2',\n",
       " 'ToxCast_TOX21_p53_BLA_p1_ratio',\n",
       " 'ToxCast_TOX21_p53_BLA_p1_viability',\n",
       " 'ToxCast_TOX21_p53_BLA_p2_ch1',\n",
       " 'ToxCast_TOX21_p53_BLA_p2_ch2',\n",
       " 'ToxCast_TOX21_p53_BLA_p2_ratio',\n",
       " 'ToxCast_TOX21_p53_BLA_p2_viability',\n",
       " 'ToxCast_TOX21_p53_BLA_p3_ch1',\n",
       " 'ToxCast_TOX21_p53_BLA_p3_ch2',\n",
       " 'ToxCast_TOX21_p53_BLA_p3_ratio',\n",
       " 'ToxCast_TOX21_p53_BLA_p3_viability',\n",
       " 'ToxCast_TOX21_p53_BLA_p4_ch1',\n",
       " 'ToxCast_TOX21_p53_BLA_p4_ch2',\n",
       " 'ToxCast_TOX21_p53_BLA_p4_ratio',\n",
       " 'ToxCast_TOX21_p53_BLA_p4_viability',\n",
       " 'ToxCast_TOX21_p53_BLA_p5_ch1',\n",
       " 'ToxCast_TOX21_p53_BLA_p5_ch2',\n",
       " 'ToxCast_TOX21_p53_BLA_p5_ratio',\n",
       " 'ToxCast_TOX21_p53_BLA_p5_viability',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_AXIS_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_ActivityScore',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_BRAI_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_CFIN_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_CIRC_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_EYE_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_JAW_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_MORT_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_OTIC_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_PE_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_PFIN_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_PIG_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_SNOU_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_SOMI_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_SWIM_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_TRUN_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_TR_up',\n",
       " 'ToxCast_Tanguay_ZF_120hpf_YSE_up',\n",
       " 'Caco2_Wang',\n",
       " 'VDss_Lombardo',\n",
       " 'Half_Life_Obach',\n",
       " 'Clearance_Hepatocyte_AZ',\n",
       " 'Clearance_Microsome_AZ',\n",
       " 'LD50_Zhu',\n",
       " 'PPBR_AZ',\n",
       " 'BindingDB_kd',\n",
       " 'BindingDB_ic50',\n",
       " 'BindingDB_ki',\n",
       " 'DAVIS',\n",
       " 'USPTO_Yields',\n",
       " 'Buchwald_Hartwig',\n",
       " 'OncoPolyPharmacology',\n",
       " 'DrugComb_CSS',\n",
       " 'DrugComb_HSA',\n",
       " 'DrugComb_Loewe',\n",
       " 'DrugComb_Bliss',\n",
       " 'DrugComb_ZIP',\n",
       " 'Protein_SAbDab',\n",
       " 'Lipophilicity_AstraZeneca',\n",
       " 'Solubility_AqSolDB',\n",
       " 'TAP_CDR_Length',\n",
       " 'TAP_PSH',\n",
       " 'TAP_PPC',\n",
       " 'TAP_PNC',\n",
       " 'TAP_SFvCSP',\n",
       " 'Leenay_Fraction_Insertions',\n",
       " 'Leenay_Avg_Insertion_Length',\n",
       " 'Leenay_Avg_Deletion_Length',\n",
       " 'Leenay_Indel_Diversity',\n",
       " 'Leenay_Fraction_Frameshifts',\n",
       " 'KIBA',\n",
       " 'BindingDB_Patent',\n",
       " 'USPTO']"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 18
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-21T01:55:47.772618Z",
     "start_time": "2025-10-21T01:55:47.767592Z"
    }
   },
   "cell_type": "code",
   "source": "len([name for name in list(tdc_prompts_json.keys()) if not (name.startswith('ToxCast') or name.startswith('Tox21'))])",
   "id": "aca0c5b12d07dfd",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "74"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 35
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-16T19:47:55.786017Z",
     "start_time": "2025-10-16T19:47:55.783929Z"
    }
   },
   "cell_type": "code",
   "source": [
    "from tdc.utils import retrieve_dataset_names\n",
    "\n",
    "# 例：ADME 任务下的所有数据集名称\n",
    "names = retrieve_dataset_names('Tox')\n",
    "print(names)"
   ],
   "id": "7b4ae8433f971bdc",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['tox21', 'toxcast', 'clintox', 'herg_karim', 'herg', 'herg_central', 'dili', 'skin_reaction', 'ames', 'carcinogens_lagunin', 'ld50_zhu']\n"
     ]
    }
   ],
   "execution_count": 6
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-08T03:47:07.359788Z",
     "start_time": "2025-10-08T03:47:07.310530Z"
    }
   },
   "cell_type": "code",
   "source": [
    "from tdc import metadata\n",
    "from tdc.utils import retrieve_dataset_names\n",
    "\n",
    "task2cat = metadata.get_task2category()   # {task_name: category, ...}\n",
    "print(task2cat, '\\n')\n",
    "for task in sorted(task2cat.keys()):\n",
    "    ds = retrieve_dataset_names(task)\n",
    "    print(f\"[{task2cat[task]}] {task}: {ds}\\n\")"
   ],
   "id": "1d437b56b10081d0",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Tox': 'single_pred', 'ADME': 'single_pred', 'HTS': 'single_pred', 'Epitope': 'single_pred', 'Develop': 'single_pred', 'QM': 'single_pred', 'Paratope': 'single_pred', 'Yields': 'single_pred', 'CRISPROutcome': 'single_pred', 'DTI': 'multi_pred', 'PPI': 'multi_pred', 'DDI': 'multi_pred', 'PeptideMHC': 'multi_pred', 'DrugRes': 'multi_pred', 'AntibodyAff': 'multi_pred', 'DrugSyn': 'multi_pred', 'MTI': 'multi_pred', 'GDA': 'multi_pred', 'Catalyst': 'multi_pred', 'TCR_Epitope_Binding': 'multi_pred', 'TrialOutcome': 'multi_pred', 'CellXGene': 'multi_pred', 'RetroSyn': 'generation', 'Reaction': 'generation', 'MolGen': 'generation'} \n",
      "\n",
      "[single_pred] ADME: ['lipophilicity_astrazeneca', 'solubility_aqsoldb', 'hydrationfreeenergy_freesolv', 'caco2_wang', 'pampa_ncats', 'approved_pampa_ncats', 'hia_hou', 'pgp_broccatelli', 'bioavailability_ma', 'vdss_lombardo', 'cyp2c19_veith', 'cyp2d6_veith', 'cyp3a4_veith', 'cyp1a2_veith', 'cyp2c9_veith', 'cyp2c9_substrate_carbonmangels', 'cyp2d6_substrate_carbonmangels', 'cyp3a4_substrate_carbonmangels', 'bbb_martins', 'b3db_classification', 'b3db_regression', 'ppbr_az', 'half_life_obach', 'clearance_hepatocyte_az', 'clearance_microsome_az', 'hlm', 'rlm']\n",
      "\n",
      "[multi_pred] AntibodyAff: ['protein_sabdab']\n",
      "\n",
      "[single_pred] CRISPROutcome: ['leenay']\n",
      "\n",
      "[multi_pred] Catalyst: ['uspto_catalyst']\n",
      "\n",
      "[multi_pred] CellXGene: ['scperturb_drug_AissaBenevolenskaya2021', 'scperturb_drug_SrivatsanTrapnell2020_sciplex2', 'scperturb_drug_SrivatsanTrapnell2020_sciplex3', 'scperturb_drug_SrivatsanTrapnell2020_sciplex4', 'scperturb_drug_ZhaoSims2021', 'scperturb_gene_NormanWeissman2019', 'scperturb_gene_ReplogleWeissman2022_rpe1', 'scperturb_gene_ReplogleWeissman2022_k562_essential']\n",
      "\n",
      "[multi_pred] DDI: ['drugbank', 'twosides']\n",
      "\n",
      "[multi_pred] DTI: ['davis', 'kiba', 'bindingdb_kd', 'bindingdb_ic50', 'bindingdb_ki', 'bindingdb_patent']\n",
      "\n",
      "[single_pred] Develop: ['tap', 'sabdab_chen']\n",
      "\n",
      "[multi_pred] DrugRes: ['gdsc1', 'gdsc2']\n",
      "\n",
      "[multi_pred] DrugSyn: ['oncopolypharmacology', 'drugcomb']\n",
      "\n",
      "[single_pred] Epitope: ['iedb_jespersen', 'pdb_jespersen']\n",
      "\n",
      "[multi_pred] GDA: ['disgenet']\n",
      "\n",
      "[single_pred] HTS: ['hiv', 'sarscov2_3clpro_diamond', 'sarscov2_vitro_touret', 'orexin1_receptor_butkiewicz', 'm1_muscarinic_receptor_agonists_butkiewicz', 'm1_muscarinic_receptor_antagonists_butkiewicz', 'potassium_ion_channel_kir2.1_butkiewicz', 'kcnq2_potassium_channel_butkiewicz', 'cav3_t-type_calcium_channels_butkiewicz', 'choline_transporter_butkiewicz', 'serine_threonine_kinase_33_butkiewicz', 'tyrosyl-dna_phosphodiesterase_butkiewicz']\n",
      "\n",
      "[multi_pred] MTI: ['mirtarbase']\n",
      "\n",
      "[generation] MolGen: ['zinc', 'moses', 'chembl', 'chembl_v29']\n",
      "\n",
      "[multi_pred] PPI: ['huri']\n",
      "\n",
      "[single_pred] Paratope: ['sabdab_liberis']\n",
      "\n",
      "[multi_pred] PeptideMHC: ['mhc2_iedb_jensen', 'mhc1_iedb-imgt_nielsen']\n",
      "\n",
      "[single_pred] QM: ['qm7', 'qm7b', 'qm8', 'qm9']\n",
      "\n",
      "[generation] Reaction: ['uspto']\n",
      "\n",
      "[generation] RetroSyn: ['uspto50k', 'uspto']\n",
      "\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'TCR_Epitope_Binding'",
     "output_type": "error",
     "traceback": [
      "\u001B[31m---------------------------------------------------------------------------\u001B[39m",
      "\u001B[31mKeyError\u001B[39m                                  Traceback (most recent call last)",
      "\u001B[36mCell\u001B[39m\u001B[36m \u001B[39m\u001B[32mIn[6]\u001B[39m\u001B[32m, line 7\u001B[39m\n\u001B[32m      5\u001B[39m \u001B[38;5;28mprint\u001B[39m(task2cat, \u001B[33m'\u001B[39m\u001B[38;5;130;01m\\n\u001B[39;00m\u001B[33m'\u001B[39m)\n\u001B[32m      6\u001B[39m \u001B[38;5;28;01mfor\u001B[39;00m task \u001B[38;5;129;01min\u001B[39;00m \u001B[38;5;28msorted\u001B[39m(task2cat.keys()):\n\u001B[32m----> \u001B[39m\u001B[32m7\u001B[39m     ds = \u001B[43mretrieve_dataset_names\u001B[49m\u001B[43m(\u001B[49m\u001B[43mtask\u001B[49m\u001B[43m)\u001B[49m\n\u001B[32m      8\u001B[39m     \u001B[38;5;28mprint\u001B[39m(\u001B[33mf\u001B[39m\u001B[33m\"\u001B[39m\u001B[33m[\u001B[39m\u001B[38;5;132;01m{\u001B[39;00mtask2cat[task]\u001B[38;5;132;01m}\u001B[39;00m\u001B[33m] \u001B[39m\u001B[38;5;132;01m{\u001B[39;00mtask\u001B[38;5;132;01m}\u001B[39;00m\u001B[33m: \u001B[39m\u001B[38;5;132;01m{\u001B[39;00mds\u001B[38;5;132;01m}\u001B[39;00m\u001B[38;5;130;01m\\n\u001B[39;00m\u001B[33m\"\u001B[39m)\n",
      "\u001B[36mFile \u001B[39m\u001B[32m~/anaconda3/envs/reasonv/lib/python3.11/site-packages/tdc/utils/retrieve.py:102\u001B[39m, in \u001B[36mretrieve_dataset_names\u001B[39m\u001B[34m(name)\u001B[39m\n\u001B[32m     93\u001B[39m \u001B[38;5;28;01mdef\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34mretrieve_dataset_names\u001B[39m(name):\n\u001B[32m     94\u001B[39m \u001B[38;5;250m    \u001B[39m\u001B[33;03m\"\"\"to get all available dataset names given a task\u001B[39;00m\n\u001B[32m     95\u001B[39m \n\u001B[32m     96\u001B[39m \u001B[33;03m    Args:\u001B[39;00m\n\u001B[32m   (...)\u001B[39m\u001B[32m    100\u001B[39m \u001B[33;03m        list: a list of available datasets\u001B[39;00m\n\u001B[32m    101\u001B[39m \u001B[33;03m    \"\"\"\u001B[39;00m\n\u001B[32m--> \u001B[39m\u001B[32m102\u001B[39m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[43mdataset_names\u001B[49m\u001B[43m[\u001B[49m\u001B[43mname\u001B[49m\u001B[43m]\u001B[49m\n",
      "\u001B[31mKeyError\u001B[39m: 'TCR_Epitope_Binding'"
     ]
    }
   ],
   "execution_count": 6
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-08T03:50:26.765036Z",
     "start_time": "2025-10-08T03:50:21.403675Z"
    }
   },
   "cell_type": "code",
   "source": [
    "from tdc import utils\n",
    "from tdc.benchmark_group import admet_group\n",
    "\n",
    "# 方式A：直接取名称列表\n",
    "names = utils.retrieve_benchmark_names('ADMET_Group')\n",
    "print(names)\n",
    "\n",
    "# 方式B：实例化组对象后查看\n",
    "group = admet_group(path='SFT_data/')\n",
    "print(group.dataset_names)\n"
   ],
   "id": "7be66b4a0fece1b1",
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Downloading Benchmark Group...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['caco2_wang', 'hia_hou', 'pgp_broccatelli', 'bioavailability_ma', 'lipophilicity_astrazeneca', 'solubility_aqsoldb', 'bbb_martins', 'ppbr_az', 'vdss_lombardo', 'cyp2d6_veith', 'cyp3a4_veith', 'cyp2c9_veith', 'cyp2d6_substrate_carbonmangels', 'cyp3a4_substrate_carbonmangels', 'cyp2c9_substrate_carbonmangels', 'half_life_obach', 'clearance_microsome_az', 'clearance_hepatocyte_az', 'herg', 'ames', 'dili', 'ld50_zhu']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 1.47M/1.47M [00:00<00:00, 2.08MiB/s]\n",
      "Extracting zip file...\n",
      "Done!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['caco2_wang', 'hia_hou', 'pgp_broccatelli', 'bioavailability_ma', 'lipophilicity_astrazeneca', 'solubility_aqsoldb', 'bbb_martins', 'ppbr_az', 'vdss_lombardo', 'cyp2d6_veith', 'cyp3a4_veith', 'cyp2c9_veith', 'cyp2d6_substrate_carbonmangels', 'cyp3a4_substrate_carbonmangels', 'cyp2c9_substrate_carbonmangels', 'half_life_obach', 'clearance_microsome_az', 'clearance_hepatocyte_az', 'herg', 'ames', 'dili', 'ld50_zhu']\n"
     ]
    }
   ],
   "execution_count": 7
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-17T01:42:54.879592Z",
     "start_time": "2025-10-17T01:42:54.474066Z"
    }
   },
   "cell_type": "code",
   "source": [
    "from tdc.utils import retrieve_label_name_list\n",
    "label_list = retrieve_label_name_list('herg_central')\n",
    "print(label_list)\n",
    "from tdc.single_pred import Tox\n",
    "data = Tox(name = 'herg_central', label_name = label_list[0])\n",
    "split = data.get_split()\n",
    "print(split['test'])"
   ],
   "id": "88f62657d9b21664",
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found local copy...\n",
      "Loading...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['hERG_at_1uM', 'hERG_at_10uM', 'hERG_inhib']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        Drug_ID                                               Drug        Y\n",
      "0      17508895              Cc1c(C(=O)NCc2ccco2)cccc1[N+](=O)[O-]   9.7453\n",
      "1      49647072            CC1(C(=O)O)CCC(C(=O)Nc2cccc(F)c2)C1(C)C   2.3245\n",
      "2      49825891             N=c1oc2ccc(Br)cc2cc1C(=O)NCc1ccc(F)cc1   8.7043\n",
      "3      49649013          O=C(O)c1cc(-c2ccccc2Cl)n(-c2cccc(Cl)c2)n1  10.2614\n",
      "4      47195785         COc1cccc(OP(=O)(Nc2ccccc2)Oc2cccc(OC)c2)c1  10.3881\n",
      "...         ...                                                ...      ...\n",
      "61374  49737086      O=c1ccc(-c2ccc(S(=O)(=O)Nc3ccccc3Cl)o2)n[nH]1  13.7994\n",
      "61375  22406625            Cc1cc(Br)ccc1SCC(=O)OCC(=O)Nc1ccccc1C#N  11.7833\n",
      "61376  24821870              COc1ccc(CSC2=CS(=O)(=O)c3ccccc3N2)cc1  13.7980\n",
      "61377  56463241              O=C(/C=C/C=C/c1ccc2c(c1)OCO2)N1CCCCC1   9.1216\n",
      "61378   4244597  COc1cccc(-n2nnc3c(=O)n(CC(=O)N4CCCc5ccccc54)cn...  -6.2821\n",
      "\n",
      "[61379 rows x 3 columns]\n"
     ]
    }
   ],
   "execution_count": 5
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-10-17T01:33:16.618493Z",
     "start_time": "2025-10-17T01:33:16.420594Z"
    }
   },
   "cell_type": "code",
   "source": [
    "from tdc.single_pred import Tox\n",
    "data = Tox(name = 'ToxCast')\n",
    "split = data.get_split()"
   ],
   "id": "c5e3d6f72b400df6",
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Please select a label name. You can use tdc.utils.retrieve_label_name_list('toxcast') to retrieve all available label names.",
     "output_type": "error",
     "traceback": [
      "\u001B[31m---------------------------------------------------------------------------\u001B[39m",
      "\u001B[31mValueError\u001B[39m                                Traceback (most recent call last)",
      "\u001B[36mCell\u001B[39m\u001B[36m \u001B[39m\u001B[32mIn[2]\u001B[39m\u001B[32m, line 2\u001B[39m\n\u001B[32m      1\u001B[39m \u001B[38;5;28;01mfrom\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34;01mtdc\u001B[39;00m\u001B[34;01m.\u001B[39;00m\u001B[34;01msingle_pred\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[38;5;28;01mimport\u001B[39;00m Tox\n\u001B[32m----> \u001B[39m\u001B[32m2\u001B[39m data = \u001B[43mTox\u001B[49m\u001B[43m(\u001B[49m\u001B[43mname\u001B[49m\u001B[43m \u001B[49m\u001B[43m=\u001B[49m\u001B[43m \u001B[49m\u001B[33;43m'\u001B[39;49m\u001B[33;43mToxCast\u001B[39;49m\u001B[33;43m'\u001B[39;49m\u001B[43m)\u001B[49m\n\u001B[32m      3\u001B[39m split = data.get_split()\n",
      "\u001B[36mFile \u001B[39m\u001B[32m~/anaconda3/envs/reasonv/lib/python3.11/site-packages/tdc/single_pred/tox.py:39\u001B[39m, in \u001B[36mTox.__init__\u001B[39m\u001B[34m(self, name, path, label_name, print_stats, convert_format)\u001B[39m\n\u001B[32m     30\u001B[39m \u001B[38;5;28;01mdef\u001B[39;00m\u001B[38;5;250m \u001B[39m\u001B[34m__init__\u001B[39m(\n\u001B[32m     31\u001B[39m     \u001B[38;5;28mself\u001B[39m,\n\u001B[32m     32\u001B[39m     name,\n\u001B[32m   (...)\u001B[39m\u001B[32m     36\u001B[39m     convert_format=\u001B[38;5;28;01mNone\u001B[39;00m,\n\u001B[32m     37\u001B[39m ):\n\u001B[32m     38\u001B[39m \u001B[38;5;250m    \u001B[39m\u001B[33;03m\"\"\"Create a Tox (Toxicity Prediction) dataloader object.\"\"\"\u001B[39;00m\n\u001B[32m---> \u001B[39m\u001B[32m39\u001B[39m     \u001B[38;5;28;43msuper\u001B[39;49m\u001B[43m(\u001B[49m\u001B[43m)\u001B[49m\u001B[43m.\u001B[49m\u001B[34;43m__init__\u001B[39;49m\u001B[43m(\u001B[49m\n\u001B[32m     40\u001B[39m \u001B[43m        \u001B[49m\u001B[43mname\u001B[49m\u001B[43m,\u001B[49m\n\u001B[32m     41\u001B[39m \u001B[43m        \u001B[49m\u001B[43mpath\u001B[49m\u001B[43m,\u001B[49m\n\u001B[32m     42\u001B[39m \u001B[43m        \u001B[49m\u001B[43mlabel_name\u001B[49m\u001B[43m,\u001B[49m\n\u001B[32m     43\u001B[39m \u001B[43m        \u001B[49m\u001B[43mprint_stats\u001B[49m\u001B[43m,\u001B[49m\n\u001B[32m     44\u001B[39m \u001B[43m        \u001B[49m\u001B[43mdataset_names\u001B[49m\u001B[43m=\u001B[49m\u001B[43mdataset_names\u001B[49m\u001B[43m[\u001B[49m\u001B[33;43m\"\u001B[39;49m\u001B[33;43mTox\u001B[39;49m\u001B[33;43m\"\u001B[39;49m\u001B[43m]\u001B[49m\u001B[43m,\u001B[49m\n\u001B[32m     45\u001B[39m \u001B[43m        \u001B[49m\u001B[43mconvert_format\u001B[49m\u001B[43m=\u001B[49m\u001B[43mconvert_format\u001B[49m\u001B[43m,\u001B[49m\n\u001B[32m     46\u001B[39m \u001B[43m    \u001B[49m\u001B[43m)\u001B[49m\n\u001B[32m     47\u001B[39m     \u001B[38;5;28;01mif\u001B[39;00m print_stats:\n\u001B[32m     48\u001B[39m         \u001B[38;5;28mself\u001B[39m.print_stats()\n",
      "\u001B[36mFile \u001B[39m\u001B[32m~/anaconda3/envs/reasonv/lib/python3.11/site-packages/tdc/single_pred/single_pred_dataset.py:65\u001B[39m, in \u001B[36mDataLoader.__init__\u001B[39m\u001B[34m(self, name, path, label_name, print_stats, dataset_names, convert_format, raw_format)\u001B[39m\n\u001B[32m     63\u001B[39m \u001B[38;5;28;01mif\u001B[39;00m name.lower() \u001B[38;5;129;01min\u001B[39;00m dataset2target_lists.keys():\n\u001B[32m     64\u001B[39m     \u001B[38;5;28;01mif\u001B[39;00m label_name \u001B[38;5;129;01mis\u001B[39;00m \u001B[38;5;28;01mNone\u001B[39;00m:\n\u001B[32m---> \u001B[39m\u001B[32m65\u001B[39m         \u001B[38;5;28;01mraise\u001B[39;00m \u001B[38;5;167;01mValueError\u001B[39;00m(\n\u001B[32m     66\u001B[39m             \u001B[33m\"\u001B[39m\u001B[33mPlease select a label name. You can use tdc.utils.retrieve_label_name_list(\u001B[39m\u001B[33m'\u001B[39m\u001B[33m\"\u001B[39m\n\u001B[32m     67\u001B[39m             + name.lower() +\n\u001B[32m     68\u001B[39m             \u001B[33m\"\u001B[39m\u001B[33m'\u001B[39m\u001B[33m) to retrieve all available label names.\u001B[39m\u001B[33m\"\u001B[39m)\n\u001B[32m     70\u001B[39m entity1, y, entity1_idx = property_dataset_load(name, path, label_name,\n\u001B[32m     71\u001B[39m                                                 dataset_names)\n\u001B[32m     73\u001B[39m \u001B[38;5;28mself\u001B[39m.entity1 = entity1\n",
      "\u001B[31mValueError\u001B[39m: Please select a label name. You can use tdc.utils.retrieve_label_name_list('toxcast') to retrieve all available label names."
     ]
    }
   ],
   "execution_count": 2
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": "",
   "id": "d77fcd354fe345ed"
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
